---
title: "Code_TestCorrelations"
author: "Edna Chiang"
date: "12/14/2021"
output: html_document
---

### Load Packages, scripts, and functions
```{r}
library(ggplot2)
library(plyr)
library(dplyr)
library(grid)
library(car)
library(confintr)
source('scripts/misc.R')
set.seed(1)
theme_set(theme_bw())
```

### Load Data
```{r}
load('crds.norm.RData')
load('crds.use.Rdata')
meta <- read.csv('metadata.csv')
```

### Prepare CRDS data
```{r}
# Add metadata
crds.use[, colnames(meta)[3:ncol(meta)]] <- NA
for(i in 1:nrow(crds.use)){
  meta.row <- which(meta$ID == crds.use$ID[i])
  crds.use[i, 7:ncol(crds.use)] <- meta[meta.row, 3:ncol(meta)]
}

# Calculate square root of AUC
crds.use$AUC <- as.numeric(crds.use$AUC)
minAUC <- min(crds.use$AUC)
crds.use$AUC_sqrt <- crds.use$AUC + (minAUC * -1)
crds.use$AUC_sqrt <- sqrt(crds.use$AUC_sqrt)

# Convert to numeric
crds.use$AvgD <- as.numeric(crds.use$AvgD)
crds.use$MaxD <- as.numeric(crds.use$MaxD)
crds.use$MaxSlope <- as.numeric(crds.use$MaxSlope)
crds.use$AUC <- as.numeric(crds.use$AUC)
crds.use$MaxD <- as.numeric(crds.use$MaxD)
crds.use$Baseline <- as.numeric(crds.use$Baseline)

# Order categorical variables
crds.use$Season <- ordered(crds.use$Season, levels = c("Summer", "Winter", "Spring"))
crds.use$Sex <- ordered(crds.use$Sex, levels = c("Female", "Male"))
crds.use$Antibiotics <- ordered(crds.use$Antibiotics, levels = c("No", "Yes"))
crds.use$Substrate <- ordered(crds.use$Substrate, levels = c("Inulin", "Saline"))
crds.use$Diet <- ordered(crds.use$Diet, levels = c("1", "2"))

# Remove outliers
crds.rem <- crds.use[-which(crds.use$ID == "S23"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "S37"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "S60"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "W43"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "W20"),]
  # This is due to odd metagenome
crds.rem <- crds.rem[-which(crds.rem$ID == "W50"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "P20"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "P22"),]


# Separate by substrate
crds.inu <- crds.rem[which(crds.rem$Substrate == "Inulin"),]
crds.sal <- crds.rem[which(crds.rem$Substrate == "Saline"),]

# Separate by substrate + season
crds.inu.s <- crds.inu[which(crds.inu$Season == "Summer"),]
crds.inu.w <- crds.inu[which(crds.inu$Season == "Winter"),]
crds.inu.p <- crds.inu[which(crds.inu$Season == "Spring"),]
crds.sal.s <- crds.sal[which(crds.sal$Season == "Summer"),]
crds.sal.w <- crds.sal[which(crds.sal$Season == "Winter"),]
crds.sal.p <- crds.sal[which(crds.sal$Season == "Spring"),]

# Separate by substrate + ABX
crds.inu.noabx <- crds.inu[which(crds.inu$Antibiotics == "No"),]
crds.inu.abx <- crds.inu[which(crds.inu$Antibiotics == "Yes"),]
crds.sal.noabx <- crds.sal[which(crds.sal$Antibiotics == "No"),]
crds.sal.abx <- crds.sal[which(crds.sal$Antibiotics == "Yes"),]
```

### Decide variables
```{r}
res <- resid(lm(MaxD ~ MaxSlope, crds.rem))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 1.481e-07
  # Not Normal
cor.test(crds.rem$MaxD, crds.rem$MaxSlope, method = "kendall")
  # p = < 2.2e-16
  # tau = 0.7378749

res <- resid(lm(Baseline ~ MaxD, crds.rem))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 0.0008238
  # Not Normal
cor.test(crds.rem$Baseline, crds.rem$MaxD, method = "kendall")
  # p = 0.4694
  # tau = -0.05742869


res <- resid(lm(Baseline ~ MaxSlope, crds.rem))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 0.0007673
  # Not Normal
cor.test(crds.rem$Baseline, crds.rem$MaxSlope, method = "kendall")
  # p = 0.2728
  # tau = -0.08706931


res <- resid(lm(BodyMass ~ BodyTemp, crds.rem))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
cor.test(crds.rem$BodyMass, crds.rem$BodyTemp, method = "pearson")
  # p = 0.0199
  # cor = -0.2701917
res <- resid(lm(BodyMass ~ Ketones, crds.rem))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
cor.test(crds.rem$BodyMass, crds.rem$Ketones, method = "pearson")
  # p = 8.288e-08
  # cor = -0.6313116
res <- resid(lm(BodyTemp ~ Ketones, crds.rem))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
cor.test(crds.rem$BodyTemp, crds.rem$Ketones, method = "pearson")
  # p = 0.2755
  # cor = -0.1443226

## Go with Baseline, MaxSlope, & BodyMass
```

### Test Baseline
```{r}
### Test BodyMass Correlation

# Inulin No ABX
res <- resid(lm(BodyMass ~ Baseline, crds.inu.noabx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 0.04097
  # Not Normal
cor.test(crds.inu.noabx$BodyMass, crds.inu.noabx$Baseline, method = "kendall")
  # p = 0.0004751
  # tau = 0.5333333
ci_cor(crds.inu.noabx$BodyMass, crds.inu.noabx$Baseline, method = "kendall", type = c("bootstrap"), R = 9999, boot_type = "bca")
  # 0.122449 0.760000


# Inulin ABX
res <- resid(lm(BodyMass ~ Baseline, crds.inu.abx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(crds.inu.abx$BodyMass, crds.inu.abx$Baseline, method = "pearson")
  # p = 0.02593
  # CI = 0.07432002 0.79562589
  # cor = 0.5230417


# Saline No ABX
res <- resid(lm(BodyMass ~ Baseline, crds.sal.noabx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(crds.sal.noabx$BodyMass, crds.sal.noabx$Baseline, method = "pearson")
  # p = 0.1602
  # CI = -0.1399968  0.6853107
  # cor = 0.3355493 


# Saline ABX
res <- resid(lm(BodyMass ~ Baseline, crds.sal.abx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Test says normal, but residuals are sigmoid/skewed
  # Not Normal
cor.test(crds.sal.abx$BodyMass, crds.sal.abx$Baseline, method = "kendall")
  # p = 0.001143
  # tau = 0.5833333 
ci_cor(crds.sal.abx$BodyMass, crds.sal.abx$Baseline, method = "kendall", type = c("bootstrap"), R = 9999, boot_type = "bca")
  # 0.3153153 0.7818182



p.adjust(c(0.0004751, 0.02593, 0.1602, 0.001143), "BH")
  # 0.00190040 0.03457333 0.16020000 0.00228600
```


### Test MaxSlope
```{r}
### Test BodyMass Correlation

# Inulin No ABX
res <- resid(lm(BodyMass ~ MaxSlope, crds.inu.noabx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 0.05882
  # Not Normal
cor.test(crds.inu.noabx$BodyMass, crds.inu.noabx$MaxSlope, method = "kendall")
  # p = 0.007431
  # tau = 0.4190476
ci_cor(crds.inu.noabx$BodyMass, crds.inu.noabx$MaxSlope, method = "kendall", type = c("bootstrap"), R = 9999, boot_type = "bca")
  # 0.1919192 0.5959596 

# Inulin ABX
res <- resid(lm(BodyMass ~ MaxSlope, crds.inu.abx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 0.02883
  # Not normal
cor.test(crds.inu.abx$BodyMass, crds.inu.abx$MaxSlope, method = "kendall")
  # p = 0.7652
  # tau = -0.05882353 
ci_cor(crds.inu.abx$BodyMass, crds.inu.abx$MaxSlope, method = "kendall", type = c("bootstrap"), R = 9999, boot_type = "bca")
  # -0.3943662  0.3750000

# Saline No ABX
res <- resid(lm(BodyMass ~ MaxSlope, crds.sal.noabx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(crds.sal.noabx$BodyMass, crds.sal.noabx$MaxSlope, method = "pearson")
  # p = 0.9934
  # CI = -0.4526017  0.4558139
  # cor = 0.00202355 

# Saline ABX
res <- resid(lm(BodyMass ~ MaxSlope, crds.sal.abx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 0.00343
  # Only not normal because of 2 outliers
  # Normal
cor.test(crds.sal.abx$BodyMass, crds.sal.abx$MaxSlope, method = "pearson")
  # p = 0.4084
  # CI = -0.6466189  0.3074495
  # cor = -0.2221057



p.adjust(c(0.007431, 0.7652, 0.9934, 0.00343), "BH")
  #  0.014862 0.993400 0.993400 0.013720
```




### Load/Test Inulinase RPKG
```{r}
### Exo - RPKG
exo_rpkg <- read.csv('eCAMI/inulinase/exoRPKG.csv')
exo_rpkg$Season <- ordered(exo_rpkg$Season, c("Summer", "Winter", "Spring"))

# Remove W20 outlier
exo_rpkg <- exo_rpkg[-which(exo_rpkg$Sample == "S-W20"),]


### Prepare AUC & slope data
for(sample in 1:nrow(exo_rpkg)){
  sampName <- substr(exo_rpkg$Sample[sample], 3, 5)
  useRow <- which(crds.rem$ID == sampName)
  exo_rpkg$Baseline[sample] <- crds.rem$Baseline[useRow]
  exo_rpkg$AUC_sqrt[sample] <- crds.rem$AUC_sqrt[useRow]
  exo_rpkg$AvgD[sample] <- crds.rem$AvgD[useRow]
  exo_rpkg$MaxD[sample] <- crds.rem$MaxD[useRow]
  exo_rpkg$MaxSlope[sample] <- crds.rem$MaxSlope[useRow]
}

# Subset out inulin samples
exo_rpkg_inu <- exo_rpkg[1:9,]


# Test RPKG - Baseline
res <- resid(lm(Baseline ~ RPKG, exo_rpkg_inu))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Test says normal, but doesn't look like it to me
cor.test(exo_rpkg_inu$Baseline, exo_rpkg_inu$RPKG, method = "kendall")
  # p = 1
  # tau = 0
ci_cor(exo_rpkg_inu$Baseline, exo_rpkg_inu$RPKG, method = "kendall", type = c("bootstrap"), R = 9999, boot_type = "bca")
  # -0.7333333  0.5000000


# Test RPKG - MaxSlope
res <- resid(lm(MaxSlope ~ RPKG, exo_rpkg_inu))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(exo_rpkg_inu$MaxSlope, exo_rpkg_inu$RPKG, method = "pearson")
  # p = 0.9149
  # CI = -0.6868670  0.6400768
  # cor = -0.04182371
```


### Load/Test DNA Yield
```{r}
# Load DNA data
dna <- read.csv('dna.csv')
dna$Season <- ordered(dna$Season, levels=c("Summer", "Winter", "Spring"))
dna <- dna[-which(dna$Squirrel == "S23"),]
dna <- dna[-which(dna$Squirrel == "S37"),]
dna <- dna[-which(dna$Squirrel == "S60"),]
dna <- dna[-which(dna$Squirrel == "W43"),]
dna <- dna[-which(dna$Squirrel == "W20"),]
  # This is due to odd metagenome
dna <- dna[-which(dna$Squirrel == "W50"),]
dna <- dna[-which(dna$Squirrel == "P20"),]
dna <- dna[-which(dna$Squirrel == "P22"),]
dna.con <- dna[which(substr(dna$Sample,2,2) == "C"),]

# Remove samples w/o crds data
dna.con <- dna.con[-which(dna.con$Squirrel == "P04"),]
dna.con <- dna.con[-which(dna.con$Squirrel == "S11"),]
dna.con <- dna.con[-which(dna.con$Squirrel == "S40"),]


### Add in delta metrics
for(sample in 1:nrow(dna.con)){
  sampName <- dna.con$Squirrel[sample]
  useRow <- which(crds.rem$ID == sampName)
  dna.con$Baseline[sample] <- crds.rem$Baseline[useRow]
  dna.con$AUC_sqrt[sample] <- crds.rem$AUC_sqrt[useRow]
  dna.con$AvgD[sample] <- crds.rem$AvgD[useRow]
  dna.con$MaxD[sample] <- crds.rem$MaxD[useRow]
  dna.con$MaxSlope[sample] <- crds.rem$MaxSlope[useRow]
}

# Subset out inulin
dna.c.inu <- dna.con[which(dna.con$Substrate == "Inulin"),]

# Subset out ABX treatments
dna.c.inu.abx <- dna.c.inu[which(dna.c.inu$Antibiotics == "Antibiotics"),]
dna.c.inu.noabx <- dna.c.inu[which(dna.c.inu$Antibiotics == "No Antibiotics"),]



### Test Baseline
res <- resid(lm(DNA_Yield_.ng.mg. ~ Baseline, dna.c.inu.noabx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dna.c.inu.noabx$DNA_Yield_.ng.mg., dna.c.inu.noabx$Baseline, method = "pearson")
  # p = 0.3841
  # CI = -0.5816553  0.2533366
  # cor = -0.2002499 

res <- resid(lm(DNA_Yield_.ng.mg. ~ Baseline, dna.c.inu.abx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 0.01664
  # Not Normal
cor.test(dna.c.inu.abx$DNA_Yield_.ng.mg., dna.c.inu.abx$Baseline, method = "kendall")
  # p = 0.09472
  # tau = -0.2904433
ci_cor(dna.c.inu.abx$DNA_Yield_.ng.mg., dna.c.inu.abx$Baseline, method = "kendall", type = c("bootstrap"), R = 9999, boot_type = "bca")
  # -0.6275368  0.1882204


### Test MaxSlope
res <- resid(lm(DNA_Yield_.ng.mg. ~ MaxSlope, dna.c.inu.noabx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dna.c.inu.noabx$DNA_Yield_.ng.mg., dna.c.inu.noabx$MaxSlope, method = "pearson")
  # p = 0.3974
  # CI = -0.2586139  0.5779063
  # cor = 0.1948229

res <- resid(lm(DNA_Yield_.ng.mg. ~ MaxSlope, dna.c.inu.abx))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 0.009944
  # Not Normal
cor.test(dna.c.inu.abx$DNA_Yield_.ng.mg., dna.c.inu.abx$MaxSlope, method = "kendall")
  # p = 0.04829
  # tau = 0.3432511 
ci_cor(dna.c.inu.abx$DNA_Yield_.ng.mg., dna.c.inu.abx$MaxSlope, method = "kendall", type = c("bootstrap"), R = 9999, boot_type = "bca")
  # -0.02717427  0.55331674 




p.adjust(c(0.3841, 0.09472, 0.3974, 0.04829), "BH")
  # 0.39740 0.18944 0.39740 0.18944

```





### Combine % microbial * DNA yield
```{r}
### Load taxonomic classification
load("kaiju/all_shortreads/use/physeq/phylum.RData")


### Calculate % microbial
phy.microbial <- subset_taxa(physeq, ta1 != "Viruses")
phy.microbial <- subset_taxa(phy.microbial, ta1 != "Unclassified")
microbe.ra <- data.frame(matrix(nrow = 18, ncol = 1))
colnames(microbe.ra) <- "%_Microbial"
microbe.otu <- data.frame(otu_table(phy.microbial))
rownames(microbe.ra) <- colnames(microbe.otu)

for(i in 1:ncol(microbe.otu)) {
  sum.microbe <- sum(microbe.otu[,i])
  microbe.ra[i,] <- sum.microbe
}

### Prep DNA yield
dna.c.noabx <- dna.con[which(dna.con$Antibiotics == "No Antibiotics"),]

### Load % Host DNA from Short Reads
host.dna <- read.csv('hostDNA_shortreads.csv')
colnames(host.dna) <- c("Sample", "%_Host")


### Combine % microbial and DNA yield
microbe.ra$Squirrel <- substr(rownames(microbe.ra),3,5)
microbe.ra$Host_Percentage <- (host.dna$`%_Host` / 100)
microbe.ra$DNAyield <- NA
microbe.ra$Season <- NA
#microbe.ra$Metabolism <- NA
microbe.ra <- microbe.ra[which(microbe.ra$Squirrel != "W20"),]

for(j in 1:nrow(microbe.ra)){
  row <- which(dna.c.noabx$Squirrel == microbe.ra$Squirrel[j])
  microbe.ra$DNAyield[j] <- dna.c.noabx$DNA_Yield_.ng.mg.[row]
  microbe.ra$Season[j] <- as.character(dna.c.noabx$Season[row])
  #microbe.ra$Metabolism[j] <- as.character(dna.c.noabx$Metabolism[row])
  rpkg_id <- substr(exo_rpkg$Sample, 3, 5)
  row2 <- which(rpkg_id == microbe.ra$Squirrel[j])
  microbe.ra$Baseline[j] <- exo_rpkg$Baseline[row2]
  microbe.ra$MaxSlope[j] <- exo_rpkg$MaxSlope[row2]
}
microbe.ra$Season <- ordered(microbe.ra$Season, c("Summer", "Winter", "Spring"))
#microbe.ra$Metabolism <- ordered(microbe.ra$Metabolism, c("Active", "Hibernation"))

microbe.ra$NonHost_DNA_Yield <- (1 - microbe.ra$Host_Percentage) * microbe.ra$DNAyield

microbe.ra$Microbial_DNA_Yield <- (microbe.ra$`%_Microbial` / 100) * microbe.ra$NonHost_DNA_Yield


```

### Test microbial DNA yield
```{r}
# Test Microbial DNA Yield ~ Baseline
res <- resid(lm(Microbial_DNA_Yield ~ Baseline, microbe.ra))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 0.726
  # Normal
cor.test(microbe.ra$Microbial_DNA_Yield, microbe.ra$Baseline,  method = "pearson")
  # p = 0.4938
  # CI = -0.6068735  0.3307656
  # cor = -0.1782114



# Test Microbial DNA Yield ~ MaxSlope
res <- resid(lm(Microbial_DNA_Yield ~ MaxSlope, microbe.ra))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(microbe.ra$Microbial_DNA_Yield, microbe.ra$MaxSlope, method = "pearson")
  # p =  0.1426
  # CI = -0.1334229  0.7227705
  # cor = 0.3710147





### Adjust p for all Inulin MetaG tests
p.adjust(c(1, 0.9149, 0.4938, 0.1426), "BH")
  # Baseline ~ RPKG = 1.0000 
  # MaxSlope ~ RPKG = 1.0000
  # Microbial DNA Yield ~ Baseline = 0.9876 
  # Microbial DNA Yield ~ MaxSlope = 0.5704

```

### Hib Dates
```{r}
### Load data ###

# Dates
dates <- read.csv('metadata_dates.csv')
dates_use <- dates[-which(dates$Season == "Summer"),]
rownames(dates_use) <- 1:nrow(dates_use)
dates_use <- dates_use[-13:-28,]
rownames(dates_use) <- 1:nrow(dates_use)
dates_use <- dates_use[-33:-40,]
dates_use <- dates_use[-3,]
  # Remove outlier P04
#dates_use <- dates_use[-which(dates_use$ID == "W43"),]
dates_use <- dates_use[-which(dates_use$ID == "W20"),]
  # This is due to odd metagenome
#dates_use <- dates_use[-which(dates_use$ID == "W50"),]
#dates_use <- dates_use[-which(dates_use$ID == "P20"),]
#dates_use <- dates_use[-which(dates_use$ID == "P22"),]



# Add CRDS metrics
for(k in 1:nrow(dates_use)){
  ID <- dates_use$ID[k]
  rowUse <- which(crds.rem$ID == ID)
  dates_use$AvgD[k] <- crds.rem$AvgD[rowUse]
  dates_use$MaxD[k] <- crds.rem$MaxD[rowUse]
  dates_use$MaxSlope[k] <- crds.rem$MaxSlope[rowUse]
  dates_use$AUC_sqrt[k] <- crds.rem$AUC_sqrt[rowUse]
  dates_use$Baseline[k] <- crds.rem$Baseline[rowUse]
}



# Separate by Substrate
dates.inu <- dates_use[which(dates_use$Substrate == "Inulin"),]
dates.sal <- dates_use[which(dates_use$Substrate == "Saline"),]

# Separate by Substrate + Season
dates.inu.p <- dates.inu[which(dates.inu$Season == "Spring"),]
dates.inu.w <- dates.inu[which(dates.inu$Season == "Winter"),]
dates.sal.p <- dates.sal[which(dates.sal$Season == "Spring"),]
dates.sal.w <- dates.sal[which(dates.sal$Season == "Winter"),]

# Check variables for cross-correlations
dates_use_check <- dates_use[which(dates_use$ID != "P19"),]
dates_use_check <- dates_use_check[which(dates_use_check$ID != "W41"),]
res <- resid(lm(Days_Hib ~ IBA.Hib, dates_use_check))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Test is borderline
  # p = 0.06271
  # Not Normal
cor.test(dates_use_check$Days_Hib, dates_use_check$IBA.Hib, method = "kendall")
  # p-value = 1
  # tau = 0



### Correlate Baseline ### 

# Inulin Winter
res <- resid(lm(Days_Hib ~ Baseline, dates.inu.w))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 0.04767
  # Not Normal
cor.test(dates.inu.w$Days_Hib, dates.inu.w$Baseline, method = "pearson")
  # p-value = 0.8616
  # CI = -0.7005837  0.6242007
  # cor = -0.0681881 

res <- resid(lm(IBA.Hib ~ Baseline, dates.inu.w))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.inu.w$IBA.Hib, dates.inu.w$Baseline, method = "pearson")
  # p-value = 0.6949
  # CI = -0.7416104  0.5691438
  # cor = -0.1526929 


# Inulin Spring
res <- resid(lm(Days_Hib ~ Baseline, dates.inu.p))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.inu.p$Days_Hib, dates.inu.p$Baseline, method = "pearson")
  # p-value = 0.09606
  # CI = -0.2515161  0.9870016
  # cor = 0.8106333 

res <- resid(lm(IBA.Hib ~ Baseline, dates.inu.p))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.inu.p$IBA.Hib, dates.inu.p$Baseline, method = "pearson")
  # p-value = 0.5982
  # CI = -0.7829395  0.9377288
  # cor = 0.3211981 


# Saline Winter
res <- resid(lm(Days_Hib ~ Baseline, dates.sal.w))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.sal.w$Days_Hib, dates.sal.w$Baseline, method = "pearson")
  # p-value = 5.622e-05
  # CI = -0.9858039 -0.7568553
  # cor = -0.9390097

res <- resid(lm(IBA.Hib ~ Baseline, dates.sal.w))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.sal.w$IBA.Hib, dates.sal.w$Baseline, method = "pearson")
  # p-value = 0.9251
  # CI = -0.6842065  0.6430302
  # cor = -0.03681214 



# Saline Spring
res <- resid(lm(Days_Hib ~ Baseline, dates.sal.p))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.sal.p$Days_Hib, dates.sal.p$Baseline, method = "pearson")
  # p-value = 0.172
  # CI = -0.3585561  0.9551933
  # cor = 0.6389266 

res <- resid(lm(IBA.Hib ~ Baseline, dates.sal.p))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.sal.p$IBA.Hib, dates.sal.p$Baseline, method = "pearson")
  # p-value = 0.6928
  # CI = -0.8134565  0.9267202
  # cor = 0.2437349 



# Days_Hib ~ Baseline
p.adjust(c(0.8616, 0.09606, 5.622e-05, 0.172), "BH")
  # 0.86160000 0.19212000 0.00022488 0.22933333

# IBA.Hib ~ Baseline
p.adjust(c(0.6949, 0.5982, 0.9251, 0.6928), "BH")
  # 0.9251 0.9251 0.9251 0.9251



### Correlate MaxSlope ### 

# Inulin Winter
res <- resid(lm(Days_Hib ~ MaxSlope, dates.inu.w))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # p = 0.00228
  # Not Normal
cor.test(dates.inu.w$Days_Hib, dates.inu.w$MaxSlope, method = "kendall")
  # p-value = 0.3454
  # tau = 0.2535463
ci_cor(dates.inu.w$Days_Hib, dates.inu.w$MaxSlope, method = "kendall", type = c("bootstrap"), R = 9999, boot_type = "bca")
  # -0.4951318  0.7500000  

res <- resid(lm(IBA.Hib ~ MaxSlope, dates.inu.w))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.inu.w$IBA.Hib, dates.inu.w$MaxSlope, method = "pearson")
  # p-value = 0.4357
  # CI = -0.8032540  0.4562669
  # cor = -0.2982206



# Inulin Spring
res <- resid(lm(Days_Hib ~ MaxSlope, dates.inu.p))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.inu.p$Days_Hib, dates.inu.p$MaxSlope, method = "pearson")
  # p-value = 0.7633
  # CI = -0.9178319  0.8326501
  # cor = -0.1869628

res <- resid(lm(IBA.Hib ~ MaxSlope, dates.inu.p))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.inu.p$IBA.Hib, dates.inu.p$MaxSlope, method = "pearson")
  # p-value = 0.6705
  # CI = -0.9293996  0.8068176
  # cor = -0.2618188


# Saline Winter
res <- resid(lm(Days_Hib ~ MaxSlope, dates.sal.w))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.sal.w$Days_Hib, dates.sal.w$MaxSlope, method = "pearson")
  # p-value = 0.6449
  # CI = -0.7207940  0.5170449
  # cor = -0.1669221  

res <- resid(lm(IBA.Hib ~ MaxSlope, dates.sal.w))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.sal.w$IBA.Hib, dates.sal.w$MaxSlope, method = "pearson")
  # p-value = 0.5894
  # CI = -0.5284451  0.7667193
  # cor = 0.2090394



# Saline Spring
res <- resid(lm(Days_Hib ~ MaxSlope, dates.sal.p))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.sal.p$Days_Hib, dates.sal.p$MaxSlope, method = "pearson")
  # p-value = 0.5546
  # CI = -0.6722882  0.8953492
  # cor = 0.3065009 

res <- resid(lm(IBA.Hib ~ MaxSlope, dates.sal.p))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
  # Normal
cor.test(dates.sal.p$IBA.Hib, dates.sal.p$MaxSlope, method = "pearson")
  # p-value = 0.9184
  # CI =  -0.8957114  0.8672095
  # cor = -0.0641018



# Days_Hib ~ MaxSlope
p.adjust(c(0.3454, 0.7633, 0.6449, 0.5546), "BH")
  # 0.7633 0.7633 0.7633 0.7633

# IBA.Hib ~ MaxSlope
p.adjust(c(0.4357, 0.6705, 0.5894, 0.9184), "BH")
  # 0.8940 0.8940 0.8940 0.9184






### Combine RPKG + Hib Dates ###
rpkg.dates <- exo_rpkg[which(exo_rpkg$Season != "Summer"),]

for(m in 1:nrow(rpkg.dates)){
  ID <- substr(rpkg.dates$Sample, 3, 5)[m]
  
  useRow <- which(dates_use$ID == ID)
  rpkg.dates$Days_Hib[m] <- dates_use$Days_Hib[useRow]
  rpkg.dates$IBA.Hib[m] <- dates_use$IBA.Hib[useRow]
  rpkg.dates$Num_IBA[m] <- dates_use$Num_IBA[useRow]
}

# Separate by Season
rpkg.dates.w <- rpkg.dates[which(rpkg.dates$Season == "Winter"),]
rpkg.dates.p <- rpkg.dates[which(rpkg.dates$Season == "Spring"),]



### Correlate RPKG w/ Hib Dates ###
res <- resid(lm(Days_Hib ~ RPKG, rpkg.dates.w))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
cor.test(rpkg.dates.w$Days_Hib, rpkg.dates.w$RPKG, method = "pearson")
  # p-value = 0.5756
  # CI = -0.9402272  0.7746305
  # cor = -0.3400072 

res <- resid(lm(IBA.Hib ~ RPKG, rpkg.dates.w))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
cor.test(rpkg.dates.w$IBA.Hib, rpkg.dates.w$RPKG, method = "pearson")
  # p-value = 0.6959
  # CI = -0.9263363  0.8143722
  # cor = -0.2411816 


res <- resid(lm(Days_Hib ~ RPKG, rpkg.dates.p))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
cor.test(rpkg.dates.p$Days_Hib, rpkg.dates.p$RPKG, method = "pearson")
  # p-value = 0.3231
  # CI = -0.5332439  0.9313537
  # cor = 0.490652 

res <- resid(lm(IBA.Hib ~ RPKG, rpkg.dates.p))
hist(res)
qqnorm(res)
qqline(res)
shapiro.test(res)
cor.test(rpkg.dates.p$IBA.Hib, rpkg.dates.p$RPKG, method = "pearson")
  # p-value = 0.8018
  # CI = -0.8525207  0.7606957
  # cor = -0.1329302 



p.adjust(c(0.5756, 0.6959, 0.3231, 0.8018), "BH")
  # 0.8018 0.8018 0.8018 0.8018

```



### Test Sex
```{r}
### Body Mass ###

#summary(aov(BodyMass ~ Season + Diet + Season*Antibiotics*Substrate*Sex, crds.rem))
#TukeyHSD(aov(BodyMass ~ Season + Diet + Season*Antibiotics*Substrate*Sex, crds.rem), conf.level = 0.95)

# Separate by season
crds.s <- crds.rem[which(crds.rem$Season == "Summer"),]
crds.w <- crds.rem[which(crds.rem$Season == "Winter"),]
crds.p <- crds.rem[which(crds.rem$Season == "Spring"),]

# No ABX
t.test(BodyMass ~ Sex, crds.s[which(crds.s$Antibiotics == "No"),])
  # p = 0.1478
t.test(BodyMass ~ Sex, crds.w[which(crds.w$Antibiotics == "No"),])
  # p = 0.02876
t.test(BodyMass ~ Sex, crds.p[which(crds.p$Antibiotics == "No"),])
  # p = 0.003547

# ABX
t.test(BodyMass ~ Sex, crds.s[which(crds.s$Antibiotics == "Yes"),])
  # p = 0.6437
t.test(BodyMass ~ Sex, crds.w[which(crds.w$Antibiotics == "Yes"),])
  # p = 0.1732
t.test(BodyMass ~ Sex, crds.p[which(crds.p$Antibiotics == "Yes"),])
  # p = 0.8766

p.adjust(c(0.1478, 0.02876, 0.003547, 0.6437, 0.1732, 0.8766), method = "BH")
  # Summer No ABX - 0.259800
  # Winter No ABX - 0.086280~
  # Spring No ABX - 0.021282*
  # Summer ABX - 0.772440
  # Winter ABX - 0.259800
  # Spring ABX - 0.876600



### Hib Dates ###
# Separate by season
dates.w <- dates_use[which(dates_use$Season == "Winter"),]
dates.p <- dates_use[which(dates_use$Season == "Spring"),]

# No ABX - Days in Hib
t.test(Days_Hib ~ Sex, dates.w[which(dates.w$Antibiotics == "No"),])
  # p = 0.1322
t.test(Days_Hib ~ Sex, dates.p[which(dates.p$Antibiotics == "No"),])
  # p = 0.2661

# No ABX - IBA:Hib
t.test(IBA.Hib ~ Sex, dates.w[which(dates.w$Antibiotics == "No"),])
  # p = 0.2314
t.test(IBA.Hib ~ Sex, dates.p[which(dates.p$Antibiotics == "No"),])
  # p = 0.6083

# ABX - Days in Hib
t.test(Days_Hib ~ Sex, dates.w[which(dates.w$Antibiotics == "Yes"),])
  # p = 0.7635
#t.test(Days_Hib ~ Sex, dates.p[which(dates.p$Antibiotics == "Yes"),])
  # Not enough samples

# ABX - IBA:Hib
t.test(IBA.Hib ~ Sex, dates.w[which(dates.w$Antibiotics == "Yes"),])
  # p = 0.3131
```