---
title: "Code"
author: "Edna Chiang"
date: "February 5, 2018"
output: html_document
---

### Load Packages, scripts, and functions
```{r}
library(ggplot2)
library(plyr)
library(dplyr)
library(grid)
library(MESS)
library(pgirmess)
library(Bolstad2)
library(phyloseq)
library(miLineage)
library(psych)
library(car)
source('scripts/import_crds.R')
#source('scripts/bin_crds.R')
#source('scripts/bin_crds2.R')
source('scripts/bin_crds3.R')
source('scripts/misc.R')
set.seed(1)
theme_set(theme_bw())

### Stats Notes
# Use 'subset' in kruskal.test
# Do NOT use 'subset' in kruskalmc or pairwise.wilcox.test; it doesn't work!
```

### Load Data
```{r}
crds <- import.crds('CSV', 'metadata.csv')
```

### Bin RelTime
```{r}
#crds.bin <- bin.crds(crds)

#crds.bin <- bin.crds2(crds)
  # Only includes inulin, mannitol, and saline

crds.bin <- bin.crds3(crds)
  # Only includes inulin, mannitol, and saline
  # Using bin.crds3 because it's most accurate in displaying the data, even if the delta curve plots are a little wonkier (bins not aligned because I bin separately for each Season + ABX group)
```




### Summarize by Substrate + ABX
```{r}
### Remove Inulin outliers
crds.rem <- crds.bin[-which(crds.bin$ID == "S34"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "S37"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "P28"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "P30"),]

### Remove Mannitol outliers
crds.rem <- crds.rem[-which(crds.rem$ID == "S14"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "S58"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "W09"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "W29"),]

# Remove Saline outliers
crds.rem <- crds.rem[-which(crds.rem$ID == "W26"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "W43"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "W50"),]
crds.rem <- crds.rem[-which(crds.rem$ID == "P04"),]
  # This is an adult, not pup

crds.rem$Season <- ordered(crds.rem$Season, levels=c("Summer", "Winter", "Spring"))


# Summarize
crds.sum <- crds.rem %>%
  group_by(Substrate, Antibiotics, Season, BinTime) %>%
  summarize(Y_Mean = mean(Delta),
            Y_Median = median(Delta),
            Y_SD = sd(Delta),
            Y_SE = se(Delta),
            Y_IQR = IQR(Delta),
            X_Mean = mean(RelTime),
            X_Median = median(RelTime),
            X_SD = sd(RelTime),
            X_SE = se(RelTime),
            X_IQR = IQR(RelTime),
            SampSize = n())
  # X_Mean is different from BinTime because BinTime was calculated with ABX and non-ABX combined
    # BinTime was calculated this way to make sure datapoints for all individuals will align together on x-axis. If datapoints are misaligned that can be confusing for the audience. So binning was done this way for the clearest visualization.
    # X_Mean is calculated for ABX and non-ABX separately, so there will be some variation
crds.sum$Season <- ordered(crds.sum$Season, levels=c("Summer", "Winter", "Spring"))
```

### Separate by substrate
```{r}
inu.all <- crds.sum[which(crds.sum$Substrate == "Inulin"),]

man.all <- crds.sum[which(crds.sum$Substrate == "Mannitol"),]

sal.all <- crds.sum[which(crds.sum$Substrate == "Saline"),]
```

### Test baseline - Diet
```{r}
### Separate samples
crds.base <- crds.rem[which(crds.rem$RelTime < 0),]
base.inu <- crds.base[which(crds.base$Substrate == "Inulin"),]
base.man <- crds.base[which(crds.base$Substrate == "Mannitol"),]
base.sal <- crds.base[which(crds.base$Substrate == "Saline"),]
#base.abx <- crds.base[which(crds.base$Antibiotics == "Yes"),]
#base.noabx <- crds.base[which(crds.base$Antibiotics == "No"),]
base.inu.abx <- base.inu[which(base.inu$Antibiotics == "Yes"),]
base.man.abx <- base.man[which(base.man$Antibiotics == "Yes"),]
base.sal.abx <- base.sal[which(base.sal$Antibiotics == "Yes"),]
base.inu.noabx <- base.inu[which(base.inu$Antibiotics == "No"),]
base.man.noabx <- base.inu[which(base.inu$Antibiotics == "No"),]
base.sal.noabx <- base.inu[which(base.inu$Antibiotics == "No"),]


### Check Normality
hist(base.inu$Delta)
qqnorm(base.inu$Delta)
qqline(base.inu$Delta)
shapiro.test(base.inu$Delta)
  # p-value = 0.09578

hist(base.inu.abx$Delta)
qqnorm(base.inu.abx$Delta)
qqline(base.inu.abx$Delta)
shapiro.test(base.inu.abx$Delta)
  # p-value = 0.05011

hist(base.inu.noabx$Delta)
qqnorm(base.inu.noabx$Delta)
qqline(base.inu.noabx$Delta)
shapiro.test(base.inu.noabx$Delta)
  # p-value = 0.2615

hist(base.man$Delta)
qqnorm(base.man$Delta)
qqline(base.man$Delta)
shapiro.test(base.man$Delta)
  # p-value = 0.01743

hist(base.man.abx$Delta)
qqnorm(base.man.abx$Delta)
qqline(base.man.abx$Delta)
shapiro.test(base.man.abx$Delta)
  # p-value = 0.07965

hist(base.man.noabx$Delta)
qqnorm(base.man.noabx$Delta)
qqline(base.man.noabx$Delta)
shapiro.test(base.man.noabx$Delta)
  # p-value = 0.2615

hist(base.sal$Delta)
qqnorm(base.sal$Delta)
qqline(base.sal$Delta)
shapiro.test(base.sal$Delta)
  # p-value = 0.003509

hist(base.sal.abx$Delta)
qqnorm(base.sal.abx$Delta)
qqline(base.sal.abx$Delta)
shapiro.test(base.sal.abx$Delta)
  # p-value = 0.01176

hist(base.sal.noabx$Delta)
qqnorm(base.sal.noabx$Delta)
qqline(base.sal.noabx$Delta)
shapiro.test(base.sal.noabx$Delta)
  # p-value = 0.2615


### Test baseline - Diet
# Mannitol
wilcox.test(Delta ~ Diet, data = base.man.noabx[which(base.man.noabx$Season == "Spring"),])
  # p-value = 0.381
wilcox.test(Delta ~ Diet, data = base.man.abx[which(base.man.abx$Season == "Summer"),])
  # p-value = 0.8571

# Saline
wilcox.test(Delta ~ Diet, data = base.sal.noabx[which(base.sal.noabx$Season == "Spring"),])
  # p-value = 0.381
wilcox.test(Delta ~ Diet, data = base.sal.abx[which(base.sal.abx$Season == "Winter"),])
  # p-value = 0.8
wilcox.test(Delta ~ Diet, data = base.sal.abx[which(base.sal.abx$Season == "Summer"),])
  # p-value = 0.8571


```
### Calculate AUC
```{r}
# AUC = Area Under Curve
# Trapezoid Method

### Calculate AUC
ids <- unique(crds.rem$ID)
auc.df <- data.frame(crds.rem[1,])
auc.df <- auc.df[,-2:-3]
  # Remove Delta and CO2 cols
auc.df <- auc.df[,-13]
  # Remove BinTime col
colnames(auc.df) <- c("AUC", "Number", "ID", "Season", "Sex", "Antibiotics", "Substrate", "Diet", "Mom", "Mom_Location", "BodyMass", "BodyTemp")
auc.df$Ketones <- NA
  # Rename cols
meta <- read.csv('metadata.csv', fileEncoding="UTF-8-BOM")

#crds.rem= crds.bin
# First normalize Rel Time so 
#for (i in 2){
for (i in 1:length(ids)){
  ID <- ids[i]
  select <- crds.rem[which(crds.rem$ID == ID),]
    # Dataframe with CRDS values for 1 squirrel

  # Normalize delta value to baseline
  select$Delta <- select$Delta - select$Delta[1]
  
  # Remove baseline from AUC calculation because we care about the AUC after gavage
  # Since we already normalized delta value to baseline, we don't need that value anymore
  # Set baseline to gavage time = 0
  select[1,1] <- 0
  
  # Normalize AUC end time
  # Since length of experiment varies for each individual, we don't want experimental length to bias our AUC calculations
  # Normalize length to min time within each substrate
    # Inu = 4.0667
    # Sal = 2.21667
    # Man = 1.95
  if (select$Substrate == "Inulin") {
    max <- which(select$RelTime > 4.06667)
    select$RelTime[max] <- 4.06667
  } else if (select$Substrate == "Mannitol") {
    max <- which(select$RelTime > 1.95)
    select$RelTime[max] <- 1.95
  } else if (select$Substrate == "Saline") {
    max <- which(select$RelTime > 2.21666667)
    select$RelTime[max] <- 2.21666667
  }

  # Create auc variable
  AUC <- 0
  
  # Run rectangle method integration - less accurate
  #for (j in 1:(nrow(select)-1)){
  #  AUC <- AUC + ((select$RelTime[j+1]-select$RelTime[j])*select$Delta[j+1])
  #}
  
  # Run trapezoid method integration - more accurate
  for(j in 1:(nrow(select)-1)){
    avgH <- (select$Delta[j+1]+select$Delta[j])/2
      # Calculates average height between start and end point
    width <- (select$RelTime[j+1]-select$RelTime[j])
    AUC <- AUC + (avgH*width)
  }
    
  auc.df[i,1] <- AUC
  auc.df[i, 2:12] <- select[1, 4:14]
  
  use <- which(meta$ID == ID)
  auc.df$Ketones[i] <- meta$Ketones[use]
}

#write.csv(auc.df, "auc.csv", row.names=F)





### Subset by Substrate & ABX
auc.inu <- auc.df[which(auc.df$Substrate == "Inulin"),]
auc.man <- auc.df[which(auc.df$Substrate == "Mannitol"),]
auc.sal <- auc.df[which(auc.df$Substrate == "Saline"),]
#auc.abx <- auc.df[which(auc.df$Antibiotics == "Yes"),]
#auc.noabx <- auc.df[which(auc.df$Antibiotics == "No"),]
auc.inu.s <- auc.inu[which(auc.inu$Season == "Summer"),]
auc.inu.w <- auc.inu[which(auc.inu$Season == "Winter"),]
auc.inu.p <- auc.inu[which(auc.inu$Season == "Spring"),]
auc.man.s <- auc.man[which(auc.man$Season == "Summer"),]
auc.man.w <- auc.man[which(auc.man$Season == "Winter"),]
auc.man.p <- auc.man[which(auc.man$Season == "Spring"),]
auc.sal.s <- auc.sal[which(auc.sal$Season == "Summer"),]
auc.sal.w <- auc.sal[which(auc.sal$Season == "Winter"),]
auc.sal.p <- auc.sal[which(auc.sal$Season == "Spring"),]
auc.inu.abx <- auc.inu[which(auc.inu$Antibiotics == "Yes"),]
auc.inu.noabx <- auc.inu[which(auc.inu$Antibiotics == "No"),]
auc.man.abx <- auc.man[which(auc.man$Antibiotics == "Yes"),]
auc.man.noabx <- auc.man[which(auc.man$Antibiotics == "No"),]
auc.sal.abx <- auc.sal[which(auc.sal$Antibiotics == "Yes"),]
auc.sal.noabx <- auc.sal[which(auc.sal$Antibiotics == "No"),]


```

### AUC Stats & Plots
```{r}
### Check Normality - Inulin
hist(auc.inu$AUC)
qqnorm(auc.inu$AUC)
qqline(auc.inu$AUC)
shapiro.test(auc.inu$AUC)
  # p-value = 9.216e-07
  # Not Normal

hist(auc.inu.s$AUC)
qqnorm(auc.inu.s$AUC)
qqline(auc.inu.s$AUC)
shapiro.test(auc.inu.s$AUC)
  # p-value = 0.01841
  # Not Normal

hist(auc.inu.w$AUC)
qqnorm(auc.inu.w$AUC)
qqline(auc.inu.w$AUC)
shapiro.test(auc.inu.w$AUC)
  # p-value = 0.001438
  # Not Normal

hist(auc.inu.p$AUC)
qqnorm(auc.inu.p$AUC)
qqline(auc.inu.p$AUC)
shapiro.test(auc.inu.p$AUC)
  # p-value = 0.005546
  # Not Normal

hist(auc.inu.noabx$AUC)
qqnorm(auc.inu.noabx$AUC)
qqline(auc.inu.noabx$AUC)
shapiro.test(auc.inu.noabx$AUC)
  # p-value = 0.01623
  # Not Normal

hist(auc.inu.abx$AUC)
qqnorm(auc.inu.abx$AUC)
qqline(auc.inu.abx$AUC)
shapiro.test(auc.inu.abx$AUC)
  # p-value = 0.8572
  # Normal


### Check Normality - Mannitol
hist(auc.man$AUC)
qqnorm(auc.man$AUC)
qqline(auc.man$AUC)
shapiro.test(auc.man$AUC)
  # p-value = 7.702e-11
  # Not Normal

hist(auc.man.s$AUC)
qqnorm(auc.man.s$AUC)
qqline(auc.man.s$AUC)
shapiro.test(auc.man.s$AUC)
  # p-value = 1.593e-05
  # Not Normal

hist(auc.man.w$AUC)
qqnorm(auc.man.w$AUC)
qqline(auc.man.w$AUC)
shapiro.test(auc.man.w$AUC)
  # p-value = 0.004246
  # Not Normal

hist(auc.man.p$AUC)
qqnorm(auc.man.p$AUC)
qqline(auc.man.p$AUC)
shapiro.test(auc.man.p$AUC)
  # p-value = 2.329e-05
  # Not Normal

hist(auc.man.noabx$AUC)
qqnorm(auc.man.noabx$AUC)
qqline(auc.man.noabx$AUC)
shapiro.test(auc.man.noabx$AUC)
  # p-value = 1.351e-05
  # Not Normal

hist(auc.man.abx$AUC)
qqnorm(auc.man.abx$AUC)
qqline(auc.man.abx$AUC)
shapiro.test(auc.man.abx$AUC)
  # p-value = 0.0002997
  # Not Normal


### Check Normality - Saline
hist(auc.sal$AUC)
qqnorm(auc.sal$AUC)
qqline(auc.sal$AUC)
shapiro.test(auc.sal$AUC)
  # p-value = 0.0579
  # Normal

hist(auc.sal.s$AUC)
qqnorm(auc.sal.s$AUC)
qqline(auc.sal.s$AUC)
shapiro.test(auc.sal.s$AUC)
  # p-value = 0.01462
  # Not Normal

hist(auc.sal.w$AUC)
qqnorm(auc.sal.w$AUC)
qqline(auc.sal.w$AUC)
shapiro.test(auc.sal.w$AUC)
  # p-value = 0.4789
  # Normal

hist(auc.sal.p$AUC)
qqnorm(auc.sal.p$AUC)
qqline(auc.sal.p$AUC)
shapiro.test(auc.sal.p$AUC)
  # p-value = 0.7142
  # Normal

hist(auc.sal.noabx$AUC)
qqnorm(auc.sal.noabx$AUC)
qqline(auc.sal.noabx$AUC)
shapiro.test(auc.sal.noabx$AUC)
  # p-value = 0.8249
  # Normal

hist(auc.sal.abx$AUC)
qqnorm(auc.sal.abx$AUC)
qqline(auc.sal.abx$AUC)
shapiro.test(auc.sal.abx$AUC)
  # p-value = 0.9535
  # Normal



### Test AUC Variance - ABX vs. no ABX
# Inulin
fligner.test(AUC ~ Antibiotics, auc.inu.s)
  # p-value = 0.005841
  # adj p = 0.0262845
fligner.test(AUC ~ Antibiotics, auc.inu.w)
  # p-value = 0.01802
  # adj p = 0.0405450
fligner.test(AUC ~ Antibiotics, auc.inu.p)
  # p-value = 0.0361
  # adj p = 0.0649800

# Mannitol
fligner.test(AUC ~ Antibiotics, auc.man.s)
  # p-value = 0.003619
  # adj p = 0.0262845
fligner.test(AUC ~ Antibiotics, auc.man.w)
  # p-value = 0.205
  # adj p = 0.2514375
fligner.test(AUC ~ Antibiotics, auc.man.p)
  # p-value = 0.01802
  # adj p = 0.0405450

# Saline
fligner.test(AUC ~ Antibiotics, auc.sal.s)
  # p-value = 0.2235
  # adj p = 0.2514375
fligner.test(AUC ~ Antibiotics, auc.sal.w)
  # p-value = 0.9611
  # adj p = 0.9611000
fligner.test(AUC ~ Antibiotics, auc.sal.p)
  # p-value = 0.0888
  # adj p = 0.1332000

p.adjust(c(0.005841, 0.01802, 0.0361, 0.003619, 0.205, 0.01802, 0.2235, 0.9611, 0.0888), method = "BH")
  # Inulin = 0.0262845* 0.0405450* 0.0649800
  # Mannitol = 0.0262845* 0.2514375 0.0405450*
  # Saline = 0.2514375 0.9611000 0.1332000



### Test AUC - ABX vs. no ABX
# Inulin
wilcox.test(AUC ~ Antibiotics, data = auc.inu.s)
  # p-value = 0.002165
  # adj p = 0.0075750
wilcox.test(AUC ~ Antibiotics, data = auc.inu.w)
  # p-value = 0.09732
  # adj p = 0.1303714
wilcox.test(AUC ~ Antibiotics, data = auc.inu.p)
  # p-value = 0.002525
  # adj p = 0.0075750

# Mannitol
wilcox.test(AUC ~ Antibiotics, data = auc.man.s)
  # p-value = 0.001166
  # p adj = 0.0075750
wilcox.test(AUC ~ Antibiotics, data = auc.man.w)
  # p-value = 0.2403
  # p adj = 0.2703375
wilcox.test(AUC ~ Antibiotics, data = auc.man.p)
  # p-value = 0.01748
  # p adj = 0.0393300

# Saline
wilcox.test(AUC ~ Antibiotics, data = auc.sal.s)
  # p-value = 0.1014
  # p adj = 0.1303714
wilcox.test(AUC ~ Antibiotics, data = auc.sal.w)
  # p-value = 0.05195
  # p adj = 0.0935100
wilcox.test(AUC ~ Antibiotics, data = auc.sal.p)
  # p-value = 0.366
  # p adj = 0.3660000

p.adjust(c(0.002165, 0.09732, 0.002525, 0.001166, 0.2403, 0.01748, 0.1014, 0.05195, 0.366), method = "BH")
  # Inulin = 0.0075750 0.1303714 0.0075750
  # Mannitol = 0.0075750 0.2703375 0.0393300
  # Saline = 0.1303714 0.0935100 0.3660000



### Test AUC Variance - Seasons
fligner.test(AUC ~ Season, auc.inu.noabx)
  # p-value = 0.01405*
fligner.test(AUC ~ Season, auc.inu.noabx[which(auc.inu.noabx$Season != "Spring"),])
  # p-value = 0.7595
fligner.test(AUC ~ Season, auc.inu.noabx[which(auc.inu.noabx$Season != "Winter"),])
  # p-value = 0.02847*
fligner.test(AUC ~ Season, auc.inu.noabx[which(auc.inu.noabx$Season != "Summer"),])
  # p-value = 0.01802*
fligner.test(AUC ~ Season, auc.man.noabx)
  # p-value = 0.1365
fligner.test(AUC ~ Season, auc.sal.noabx)
  # p-value = 0.4233
fligner.test(AUC ~ Season, auc.inu.abx)
  # p-value = 0.1303
fligner.test(AUC ~ Season, auc.man.abx)
  # p-value = 0.2493
fligner.test(AUC ~ Season, auc.sal.abx)
  # p-value = 0.8241
p.adjust(c(0.01405, 0.1365, 0.4233, 0.1303, 0.2493, 0.8241), method = "fdr")
  #0.08430 0.27300 0.50796

### Test AUC - Seasons
kruskal.test(formula = AUC ~ Season, data=auc.inu.noabx)
  # p-value = 0.2006
kruskal.test(formula = AUC ~ Season, data=auc.man.noabx)
  # p-value = 0.1544
kruskal.test(formula = AUC ~ Season, data=auc.sal.noabx)
  # p-value = 0.7246
kruskal.test(formula = AUC ~ Season, data=auc.inu.abx)
  # p-value = 0.2156
kruskal.test(formula = AUC ~ Season, data=auc.man.abx)
  # p-value = 0.593
kruskal.test(formula = AUC ~ Season, data=auc.sal.abx)
  # p-value = 0.5819

p.adjust(c(0.2006, 0.1544, 0.7246, 0.2156, 0.593, 0.5819), method = "BH")
  # 0.4312 0.4312 0.7246
  # 0.4312 0.7116 0.7116


### Test AUC - Diet
# Mannitol
wilcox.test(AUC ~ Diet, data = auc.man.noabx[which(auc.man.noabx$Season == "Spring"),])
  # p-value = 0.1143
wilcox.test(AUC ~ Diet, data = auc.man.abx[which(auc.man.abx$Season == "Summer"),])
  # p-value = 0.6286

# Saline
wilcox.test(AUC ~ Diet, data = auc.sal.noabx[which(auc.sal.noabx$Season == "Spring"),])
  # p-value = 0.05714
wilcox.test(AUC ~ Diet, data = auc.sal.abx[which(auc.sal.abx$Season == "Winter"),])
  # p-value = 0.4
wilcox.test(AUC ~ Diet, data = auc.sal.abx[which(auc.sal.abx$Season == "Summer"),])
  # p-value = 0.8571



### Test AUC Variance - Diet
# Mannitol
fligner.test(AUC ~ Diet, auc.man.noabx[which(auc.man.noabx$Season == "Spring"),])
  # p-value = 0.5857
fligner.test(AUC ~ Diet, auc.man.abx[which(auc.man.abx$Season == "Summer"),])
  # p-value = 0.16839

# Saline
fligner.test(AUC ~ Diet, auc.sal.noabx[which(auc.sal.noabx$Season == "Spring"),])
  # p-value = 0.1697
fligner.test(AUC ~ Diet, auc.sal.abx[which(auc.sal.abx$Season == "Winter"),])
  # p-value = 0.1977
fligner.test(AUC ~ Diet, auc.sal.abx[which(auc.sal.abx$Season == "Summer"),])
  # p-value = 0.416




### Test AUC - BodyMass Correlation ###

# Inulin
cor.test(auc.inu.noabx$AUC, auc.inu.noabx$BodyMass, method = "spearman")
  # p-value = 0.4802
cor.test(auc.inu.s[which(auc.inu.s$Antibiotics == "No"),1],
         auc.inu.s[which(auc.inu.s$Antibiotics == "No"),11], method = "spearman")
  # p-value = 1
cor.test(auc.inu.w[which(auc.inu.w$Antibiotics == "No"),1],
         auc.inu.w[which(auc.inu.w$Antibiotics == "No"),11], method = "spearman")
  # p-value = 0.4444
cor.test(auc.inu.p[which(auc.inu.p$Antibiotics == "No"),1],
         auc.inu.p[which(auc.inu.p$Antibiotics == "No"),11], method = "spearman")
  # p-value = 0.5948


# Saline
cor.test(auc.sal.noabx$AUC, auc.sal.noabx$BodyMass, method = "spearman")
  # p-value = 0.2654
cor.test(auc.sal.s[which(auc.sal.s$Antibiotics == "No"),1],
         auc.sal.s[which(auc.sal.s$Antibiotics == "No"),11], method = "spearman")
  # p-value = 0.7139
cor.test(auc.sal.w[which(auc.sal.w$Antibiotics == "No"),1],
         auc.sal.w[which(auc.sal.w$Antibiotics == "No"),11], method = "spearman")
  # p-value = 0.3556
cor.test(auc.sal.p[which(auc.sal.p$Antibiotics == "No"),1],
         auc.sal.p[which(auc.sal.p$Antibiotics == "No"),11], method = "spearman")
  # p-value = 0.8397




### Test AUC - BodyTemp Correlation ###

# Inulin
cor.test(auc.inu.noabx$AUC, auc.inu.noabx$BodyTemp, method = "spearman")
  # p-value = 0.0576
cor.test(auc.inu.s[which(auc.inu.s$Antibiotics == "No"),1],
         auc.inu.s[which(auc.inu.s$Antibiotics == "No"),12], method = "spearman")
  # p-value = 6121
cor.test(auc.inu.w[which(auc.inu.w$Antibiotics == "No"),1],
         auc.inu.w[which(auc.inu.w$Antibiotics == "No"),12], method = "spearman")
  # p-value = 0.07433
cor.test(auc.inu.p[which(auc.inu.p$Antibiotics == "No"),1],
         auc.inu.p[which(auc.inu.p$Antibiotics == "No"),12], method = "spearman")
  # p-value = 0.3979


# Saline
cor.test(auc.sal.noabx$AUC, auc.sal.noabx$BodyTemp, method = "spearman")
  # p-value = 0.2223
cor.test(auc.sal.s[which(auc.sal.s$Antibiotics == "No"),1],
         auc.sal.s[which(auc.sal.s$Antibiotics == "No"),12], method = "spearman")
  # p-value = 0.07234
cor.test(auc.sal.w[which(auc.sal.w$Antibiotics == "No"),1],
         auc.sal.w[which(auc.sal.w$Antibiotics == "No"),12], method = "spearman")
  # p-value = 0.6483
cor.test(auc.sal.p[which(auc.sal.p$Antibiotics == "No"),1],
         auc.sal.p[which(auc.sal.p$Antibiotics == "No"),12], method = "spearman")
  # p-value = 0.4647





### Test AUC - Ketones Correlation ###

# Inulin
cor.test(auc.inu.noabx$AUC, auc.inu.noabx$Ketones, method = "spearman")
  # p-value = 0.3699
cor.test(auc.inu.s[which(auc.inu.s$Antibiotics == "No"),1],
         auc.inu.s[which(auc.inu.s$Antibiotics == "No"),13], method = "spearman")
  # p-value = NA
cor.test(auc.inu.w[which(auc.inu.w$Antibiotics == "No"),1],
         auc.inu.w[which(auc.inu.w$Antibiotics == "No"),13], method = "spearman")
  # p-value = 0.02381
  # rho = 0.8571429
  # More inulin metabolism when ketones is higher, so more inulin metab when host metabolizes more fatty acids / has low carb storage
cor.test(auc.inu.p[which(auc.inu.p$Antibiotics == "No"),1],
         auc.inu.p[which(auc.inu.p$Antibiotics == "No"),13], method = "spearman")
  # p-value = 0.6351


# Saline
cor.test(auc.sal.noabx$AUC, auc.sal.noabx$Ketones, method = "spearman")
  # p-value = 0.2355
cor.test(auc.sal.s[which(auc.sal.s$Antibiotics == "No"),1],
         auc.sal.s[which(auc.sal.s$Antibiotics == "No"),13], method = "spearman")
  # p-value = 1
cor.test(auc.sal.w[which(auc.sal.w$Antibiotics == "No"),1],
         auc.sal.w[which(auc.sal.w$Antibiotics == "No"),13], method = "spearman")
  # p-value = 0.3206
cor.test(auc.sal.p[which(auc.sal.p$Antibiotics == "No"),1],
         auc.sal.p[which(auc.sal.p$Antibiotics == "No"),13], method = "spearman")
  # p-value = 0.8007









### Prepare data for plotting
# Inulin
auc.inu$SeasonABX <- interaction(auc.inu$Season, auc.inu$Antibiotics)
auc.inu$SeasonABX <- ordered(auc.inu$SeasonABX, levels=c("Summer.No", "Summer.Yes", "Winter.No", "Winter.Yes", "Spring.No", "Spring.Yes"))

# Mannitol
auc.man$SeasonABX <- interaction(auc.man$Season, auc.man$Antibiotics)
auc.man$SeasonABX <- ordered(auc.man$SeasonABX, levels=c("Summer.No", "Summer.Yes", "Winter.No", "Winter.Yes", "Spring.No", "Spring.Yes"))

auc.sal$SeasonABX <- interaction(auc.sal$Season, auc.sal$Antibiotics)
auc.sal$SeasonABX <- ordered(auc.sal$SeasonABX, levels=c("Summer.No", "Summer.Yes", "Winter.No", "Winter.Yes", "Spring.No", "Spring.Yes"))





### AUC Dot Plots
  
# Inulin
inu.auc.plot <- ggplot(auc.inu, aes(x = Season, y = AUC, color = SeasonABX,
                                    fill = SeasonABX)) +
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize = 1.5,
               position = position_dodge(0.7), stackratio = 0.5) +
  scale_color_manual(values = c("#c98b00", "#E69F00", "#006096", "#0072B2", "#00966d",
                                "#00bf8b")) +
  scale_fill_manual(values = c("#E69F00", "white", "#0072B2", "white", "#00bf8b",
                               "white")) +
  stat_summary(fun.data = "mean_se", fun.args = list(mult = 1), geom = "pointrange",
               color = "black", position = position_dodge(0.7), size = 0.5) +
  ggtitle("Inulin")  + 
  ylab(expression("Area Under"~~delta^13~"C"~~"Curve"~"(%"[o]^2~")")) +
  scale_y_continuous(breaks=c(0, 1750, 3500, 5250, 7000, 8750), limits=c(0, 8750),
                     minor_breaks = seq(0,8750,1750)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        axis.title.x = element_text(face="bold", size=10, margin = margin(t = 7.5)),
        axis.title.y = element_text(face="bold", size=11),
        axis.text.x = element_text(size=10),
        axis.text.y = element_text(size=10),
        plot.title = element_text(hjust = 0.5, size = 14, face = "bold"),
        legend.position = "none")
  

# Mannitol  
man.auc.plot <- ggplot(auc.man, aes(x = Season, y = AUC, color = SeasonABX,
                                    fill = SeasonABX)) +
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize = 1.5,
               position = position_dodge(0.7), stackratio = 0.5) +
  scale_color_manual(values = c("#c98b00", "#E69F00", "#006096", "#0072B2", "#00966d",
                                "#00bf8b"), guide = "none") +
  scale_fill_manual(values = c("#E69F00", "white", "#0072B2", "white", "#00bf8b",
                               "white"),
                    name = "Season",
                    labels = c("Summer", "", "Winter", "", "Spring", ""),
                    guide = guide_legend(override.aes = list(fill = c("#E69F00", NA,
                                                                      "#0072B2", NA,
                                                                      "#00bf8b", NA),
                                                             alpha = c(1,0,1,0,1,0),
                                                             color = NA),
                                         keyheight = c(1.25,0,1.25,0,1.25,0))) +
  stat_summary(fun.data = "mean_se", fun.args = list(mult = 1), geom = "pointrange",
               color = "black", position = position_dodge(0.7), size = 0.5) +
  ggtitle("Mannitol") +
  #scale_y_continuous(breaks=c(0, 1750, 3500, 5250, 7000, 8750), limits=c(-7, 8750),
                     #minor_breaks = seq(0,8750,1750)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        axis.title.x = element_text(face="bold", size=10, margin = margin(t = 7.5)),
        axis.title.y = element_blank(),
        axis.text.x = element_text(size=10),
        axis.text.y = element_text(size=10),
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5, size = 14, face = "bold"),
        legend.position = "none")


# Saline
sal.auc.plot <- ggplot(auc.sal, aes(x = Season, y = AUC, color = SeasonABX,
                                    fill = SeasonABX)) +
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize = 1.5,
               position = position_dodge(0.7), stackratio = 0.5) +
  scale_color_manual(values = c("#c98b00", "#E69F00", "#006096", "#0072B2", "#00966d",
                                "#00bf8b"), guide = "none") +
  scale_fill_manual(values = c("#E69F00", "white", "#0072B2", "white", "#00bf8b",
                               "white"),
                    name = "Season",
                    labels = c("Summer", "", "Winter", "", "Spring", ""),
                    guide = guide_legend(override.aes = list(fill = c("#E69F00", NA,
                                                                      "#0072B2", NA,
                                                                      "#00bf8b", NA),
                                                             alpha = c(1,0,1,0,1,0),
                                                             color = NA),
                                         keyheight = c(1.25,0,1.25,0,1.25,0))) +
  stat_summary(fun.data = "mean_se", fun.args = list(mult = 1), geom = "pointrange",
               color = "black", position = position_dodge(0.7), size = 0.5) +
    ggtitle("Saline")  +
    theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        axis.title.x = element_text(face="bold", size=10, margin = margin(t = 7.5)),
        axis.title.y = element_blank(),
        axis.text.x = element_text(size=10),
        axis.text.y = element_text(size=10),
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5, size = 14, face = "bold"),
        legend.position = "none")












# Inulin
inu.noabx.auc.plot <- ggplot(auc.inu.noabx, aes(x = Season, y = AUC, fill = Season)) +
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize = 1.5, stackratio = 1) +
  scale_fill_manual(values = c("#E69F00", "#0072B2", "#00bf8b")) +
  stat_summary(fun.data = "mean_se", fun.args = list(mult = 1), geom = "pointrange",
               color = "black", position = position_dodge(0.7), size = 0.5) +
  ggtitle("Inulin")  + 
  ylab(expression("Area Under"~~delta^13~"C"~~"Curve"~"(%"[o]^2~")")) +
  #scale_y_continuous(breaks=c(0, 1750, 3500, 5250, 7000, 8750), limits=c(-7, 8750),
                     #minor_breaks = seq(0,8750,1750)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        axis.title.x = element_text(face="bold", size=10, margin = margin(t = 7.5)),
        axis.title.y = element_text(face="bold", size=11),
        axis.text.x = element_text(size=10),
        axis.text.y = element_text(size=10),
        plot.title = element_text(hjust = 0.5, size = 14, face = "bold"),
        legend.position = "none")
  

# Mannitol  
man.noabx.auc.plot <- ggplot(auc.man.noabx, aes(x = Season, y = AUC, fill = Season)) +
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize = 1.5, stackratio = 1) +
  scale_fill_manual(values = c("#E69F00", "#0072B2", "#00bf8b"),
                    name = "Season",
                    labels = c("Summer", "Winter", "Spring"),
                    guide = guide_legend(override.aes = list(fill = c("#E69F00",
                                                                      "#0072B2",
                                                                      "#00bf8b")))) +
  stat_summary(fun.data = "mean_se", fun.args = list(mult = 1), geom = "pointrange",
               color = "black", position = position_dodge(0.7), size = 0.5) +
  ggtitle("Mannitol") +
  #scale_y_continuous(breaks=c(0, 1750, 3500, 5250, 7000, 8750), limits=c(-7, 8750),
                     #minor_breaks = seq(0,8750,1750)) +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
        axis.title.x = element_text(face="bold", size=10, margin = margin(t = 7.5)),
        axis.title.y = element_blank(),
        axis.text.x = element_text(size=10),
        axis.text.y = element_text(size=10),
        axis.ticks.y = element_blank(),
        plot.title = element_text(hjust = 0.5, size = 14, face = "bold"),
        legend.position = "none")








##### Save AUC Dot Plots #####
#png("auc.png", width = 250, height = 75, units = "mm", res = 300)
grid.newpage()
pushViewport(viewport(layout=grid.layout(1,3, widths = c(1.1, 1.05, 0.95))))
print(inu.auc.plot, vp=viewport(layout.pos.row=1, layout.pos.col=1))
print(man.auc.plot, vp=viewport(layout.pos.row=1, layout.pos.col=2))
print(sal.auc.plot, vp=viewport(layout.pos.row=1, layout.pos.col=3))
dev.off()


#png("auc.inu.man.png", width = 175, height = 75, units = "mm", res = 300)
grid.newpage()
pushViewport(viewport(layout=grid.layout(1,3, widths = c(1.05, 0.1, 1))))
print(inu.noabx.auc.plot, vp=viewport(layout.pos.row=1, layout.pos.col=1))
print(man.noabx.auc.plot, vp=viewport(layout.pos.row=1, layout.pos.col=3))
dev.off()

```



### Summer 2020 - Inulinase Genes (metaG)
```{r}
### Import total Endo / Exo / Protein Counts
endoexo <- read.csv("eCAMI/endo_exo.csv", fileEncoding="UTF-8-BOM")
endoexo$Endo_RelAbund <- (endoexo$Endo / endoexo$Total_Protein)*100
endoexo$Exo_RelAbund <- (endoexo$Exo / endoexo$Total_Protein)*100
endoexo$Season <- c("Spring", "Spring", "Spring", "Summer", "Summer", "Summer", "Winter", "Winter", "Winter")
endoexo$Season <- ordered(endoexo$Season, levels=c("Summer", "Winter", "Spring"))

### Test Endo / Exo Rel Abund for Normality
hist(endoexo$Endo_RelAbund)
qqnorm(endoexo$Endo_RelAbund)
qqline(endoexo$Endo_RelAbund)
shapiro.test(endoexo$Endo_RelAbund)
  # p-value = 0.003571
  # Not Normal

hist(endoexo$Exo_RelAbund)
qqnorm(endoexo$Exo_RelAbund)
qqline(endoexo$Exo_RelAbund)
shapiro.test(endoexo$Exo_RelAbund)
  # p-value = 0.1865
  # Normal


### Test for Variance - Levene's Test
leveneTest(Endo_RelAbund ~ Season, data=endoexo)
  # p-value = 0.9934

leveneTest(Exo_RelAbund ~ Season, data=endoexo)
  # p-value = 0.9145


### Test Endo - Kruskal-Wallis
kruskal.test(formula = Endo_RelAbund ~ Season, data=endoexo)
  # p-value = 0.9481


### Test Exo - Kruskal-Wallis
summary(aov(Exo_RelAbund ~ Season, data=endoexo))
  # p-value = 0.901


### Import Data - Taxonomic Composition
exo.shared <- read.csv("eCAMI/exo.shared.csv", fileEncoding="UTF-8-BOM")
rownames(exo.shared) <- exo.shared$X
exo.shared <- exo.shared[,-1]
exo.tax <- read.csv("eCAMI/exo_tax.csv", fileEncoding="UTF-8-BOM")
rownames(exo.tax) <- exo.tax$Protein
exo.tax <- exo.tax[,-1]

### Make physeq object
exo.otu <- otu_table(t(exo.shared), taxa_are_rows=T)
exo.samp <- sample_data(endoexo)
exo.tax <- tax_table(exo.tax)
taxa_names(exo.tax) <- taxa_names(exo.otu)
colnames(exo.tax) <- c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")
sample_names(exo.samp) <- sample_names(exo.otu)
exo.physeq <- merge_phyloseq(exo.otu, exo.samp, exo.tax)

### Combine at diff taxa levels
exo.gen <- tax_glom(exo.physeq, taxrank="Genus")
exo.fam <- tax_glom(exo.physeq, taxrank="Family")
exo.ord <- tax_glom(exo.physeq, taxrank="Order")
exo.cla <- tax_glom(exo.physeq, taxrank="Class")



### Bray-Curtis NMDS
exo.nmds <- ordinate(physeq=exo.physeq, method="NMDS", distance="bray")
  # stress = 0.1096339

#tiff("exo.nmds.mdtp.tiff", width=8, height=6, units="in", res=600)
plot_ordination(physeq=exo.physeq, ordination=exo.nmds, color="Season") + 
  scale_color_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  geom_path(data=exo.physeq.elip.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  ggtitle("Exo-Inulinase") +
  geom_point(size=3) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.text.x = element_text(size=20),
        axis.text.y = element_text(size=20),
        axis.title.x = element_text(size=20),
        axis.title.y = element_text(size=20),
        legend.title = element_text(size=20, face="bold"),
        legend.text = element_text(size=20),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
        )
#dev.off()


### NMDS Ellipses
sample_data(exo.physeq)$Elip <- paste(sample_data(exo.physeq)$Season)
exo.physeq.elip <- data.frame(MDS1 = exo.nmds$points[,1], MDS2 = exo.nmds$points[,2],
                           Group = sample_data(exo.physeq)$Season)
plot.new()
exo.physeq.elip.ord <- ordiellipse(exo.nmds, sample_data(exo.physeq)$Elip, display="sites", kind="se", conf=0.95, label=T)
exo.physeq.elip.df <- data.frame()
for(i in levels(exo.physeq.elip$Group)){
  exo.physeq.elip.df <- rbind(exo.physeq.elip.df,
                           cbind(as.data.frame(with(exo.physeq.elip[exo.physeq.elip$Group==i,],
                                                    veganCovEllipse(exo.physeq.elip.ord[[i]]$cov,
                                                                    exo.physeq.elip.ord[[i]]$center,
                                                                    exo.physeq.elip.ord[[i]]$scale))), group=i
                                 ))
}

### Bray-Curtis PCoA
exo.pcoa <- ordinate(physeq=exo.physeq, method="PCoA", distance="bray")

plot_ordination(physeq=exo.physeq, ordination=exo.pcoa, color="Season") + 
  scale_color_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3"))



### PERMANOVA for NMDS
exo.physeq.sampdf <- data.frame(sample_data(exo.physeq))
exo.bray<- phyloseq::distance(physeq=exo.physeq, method="bray")

## Season
adonis(exo.bray~ Season, data=exo.physeq.sampdf)
  # p-value = 0.902
beta.tax = betadisper(d=exo.bray, group=exo.physeq.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.604


```
### Plot Delta Curves
```{r}
### Inulin
inu.delta.plot <- ggplot(inu.all, aes(x=BinTime, y=Y_Mean, color=Season)) +
  geom_line(aes(linetype=Antibiotics), size=1) +
   geom_point(size=3, shape=21, aes(color=Season, fill=factor(ifelse(Antibiotics == "Yes", NA, Season)))) +
  scale_color_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3"), na.value="white", guide="none") +
  geom_errorbar(aes(ymin = Y_Mean-Y_SE, ymax = Y_Mean+Y_SE), width=0.125, position=position_dodge(0.1)) +
  ggtitle(expression("Inulin"~~delta^13~"C")) +
  xlab("Time Since Gavage (hr)") +
  ylab(expression("Mean"~~delta^13~"C"~"(%"[o]~")")) +
  xlim(-0.6,4.1) +
  scale_y_continuous(breaks=c(0, 500, 1000, 1500, 2000, 2500, 3000), limits=c(-30, 3100)) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.title.x=element_text(size=20, 
                                  margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y=element_text(size=18),
        axis.text.x=element_text(size=16),
        axis.text.y=element_text(size=16),
        plot.margin = unit(c(0, 0.3, 0.3, 0.3), "cm"),
          #top, right, bottom, left
        panel.grid.major.x = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank()) +
  guides(colour=F, linetype=F)


  
## Mannitol
man.delta.plot <- ggplot(man.all, aes(x=BinTime, y=Y_Mean, color=Season)) +
  geom_line(aes(linetype=Antibiotics), size=1) +
  geom_point(size=3, shape=21, aes(color=Season, fill=factor(ifelse(Antibiotics == "Yes", NA, Season)))) +
  scale_color_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3"), na.value="white", guide="none") +
  geom_errorbar(aes(ymin = Y_Mean-Y_SE, ymax = Y_Mean+Y_SE), width=0.125, position=position_dodge(0.1)) +
  ggtitle(expression("Mannitol"~~delta^13~"C")) +
  xlab("Time Since Gavage (hr)") +
  ylab(expression("Mean"~~delta^13~"C"~"(%"[o]~")")) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.title.x=element_text(size=14),
        axis.title.y=element_text(size=14),
        axis.text.x=element_text(size=12),
        axis.text.y=element_text(size=12),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank()) +
  guides(colour=F, linetype=F)
  
  

## Saline
sal.delta.plot <- ggplot(sal.all, aes(x=BinTime, y=Y_Mean, color=Season)) +
  geom_line(aes(linetype=Antibiotics), size=1) +
  geom_point(size=3, shape=21, aes(color=Season, fill=factor(ifelse(Antibiotics == "Yes", NA, Season)))) +
  scale_color_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3"), na.value="white", guide="none") +
  geom_errorbar(aes(ymin = Y_Mean-Y_SE, ymax = Y_Mean+Y_SE), width=0.125, position=position_dodge(0.1)) +
  ggtitle(expression("Saline"~~delta^13~"C")) +
  xlab("Time Since Gavage (hr)") +
  ylab(expression("Mean"~~delta^13~"C"~"(%"[o]~")")) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.title.x=element_text(size=14),
        axis.title.y=element_text(size=14),
        axis.text.x=element_text(size=12),
        axis.text.y=element_text(size=12),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank()) +
  guides(colour=F, linetype=F)


## Steal legend
steal.legend <- ggplot(sal.all, aes(x=BinTime, y=Y_Mean, color=Season)) +
  geom_line(aes(linetype=Antibiotics), size=1) +
  geom_point(size=3) +
  scale_color_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  geom_errorbar(aes(ymin = Y_Mean-Y_SE, ymax = Y_Mean+Y_SE), width=0.125, position=position_dodge(0.1)) +
  ggtitle(expression("Saline"~~delta^13~"C")) +
  xlab("Time Since Gavage (hr)") +
  ylab(expression("Mean"~~delta^13~"C"~"(%"[o]~")")) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.title.x=element_text(size=14),
        axis.title.y=element_text(size=14),
        axis.text.x=element_text(size=12),
        axis.text.y=element_text(size=12),
        legend.title = element_text(size=20, face="bold"),
        legend.text = element_text(size=18)
        )
legend <- g_legend(steal.legend)






### All 3 substrates on same axis
carb.plot <- ggplot(crds.sum, aes(x=BinTime, y=Y_Mean, color=Season)) +
  geom_line(aes(linetype=Antibiotics), size=1) +
   geom_point(size=3, shape=21, aes(color=Season, fill=factor(ifelse(Antibiotics == "Yes", NA, Season)))) +
  scale_color_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3"), na.value="white", guide="none") +
  geom_errorbar(aes(ymin = Y_Mean-Y_SE, ymax = Y_Mean+Y_SE), width=0.125, position=position_dodge(0.1)) + 
  facet_grid(.~Substrate) +
  xlab("Time Since Gavage (hr)") +
  ylab(expression("Mean"~~delta^13~"C"~"(%"[o]~")")) +
  xlim(-0.6,4.41) +
  scale_y_continuous(breaks=c(0, 500, 1000, 1500, 2000, 2500, 3000), limits=c(-30, 3100)) +
  theme(axis.text.x = element_text(size=16),
        axis.text.y = element_text(size=16),
        axis.title.x = element_text(size=20, margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y = element_text(size=20, margin=margin(t=0, r=10, b=, l=0)),
        strip.text.x = element_text(size=20, face="bold"),
        strip.text.y = element_text(size=14),
        panel.spacing=unit(1, "lines"),
        legend.title = element_text(size = 20, face="bold"),
        legend.text = element_text(size=18),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank(),
        strip.background = element_blank()
  )



png("delta.curves.png", width = 12.75, height = 3.5, units = "in", res = 300)
carb.plot
dev.off()
```




### Plot
```{r}
## All individuals
ggplot(crds.bin, aes(x=BinTime, y=Delta, color=ID)) +
  geom_line() +
  geom_point()

## Substrate/ABX
ggplot(crds.sum, aes(x=BinTime, y=Mean, color=Substrate)) +
  geom_line(aes(linetype=Antibiotics)) +
  geom_point() +
  geom_errorbar(aes(ymin = Mean-SE, ymax = Mean+SE))

## Inulin
inu.plot <- ggplot(inu.all, aes(x=BinTime, y=Mean, color=Season)) +
  geom_line(aes(linetype=Antibiotics), size=1) +
   geom_point(size=3, shape=21, aes(color=Season, fill=factor(ifelse(Antibiotics == "Yes", NA, Season)))) +
  scale_color_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3"), na.value="white", guide="none") +
  geom_errorbar(aes(ymin = Mean-SE, ymax = Mean+SE), width=0.125, position=position_dodge(0.1)) +
  ggtitle(expression("Inulin"~~delta^13~"C")) +
  xlab("Time Since Gavage (hr)") +
  ylab(expression("Mean"~~delta^13~"C"~"(%"[o]~")")) +
  xlim(-0.6,4.1) +
  scale_y_continuous(breaks=c(0, 625, 1250, 1875, 2500), limits=c(-30, 2700)) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.title.x=element_text(size=20, 
                                  margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y=element_text(size=18),
        axis.text.x=element_text(size=16),
        axis.text.y=element_text(size=16),
        plot.margin = unit(c(0, 0.3, 0.3, 0.3), "cm"),
          #top, right, bottom, left
        panel.grid.major.x = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank()
  ) +
  guides(colour=F, linetype=F)




## Mannitol
man.plot <- ggplot(man.all, aes(x=BinTime, y=Mean, color=Season)) +
  geom_line(aes(linetype=Antibiotics), size=1) +
  geom_point(size=3, shape=21, aes(color=Season, fill=factor(ifelse(Antibiotics == "Yes", NA, Season)))) +
  scale_color_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3"), na.value="white", guide="none") +
  geom_errorbar(aes(ymin = Mean-SE, ymax = Mean+SE), width=0.125, position=position_dodge(0.1)) +
  ggtitle(expression("Mannitol"~~delta^13~"C")) +
  xlab("Time Since Gavage (hr)") +
  ylab(expression("Mean"~~delta^13~"C"~"(%"[o]~")")) +
  xlim(-0.6,2.6) +
  scale_y_continuous(breaks=c(0, 250, 500, 750, 1000), limits=c(-30, 760)) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.title.x=element_text(size=20, 
                                  margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y=element_text(size=18),
        axis.text.x=element_text(size=16),
        axis.text.y=element_text(size=16),
        plot.margin = unit(c(0, 0.3, 0.3, 0.3), "cm"),
          #top, right, bottom, left
        panel.grid.major.x = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank()
  ) +
  guides(colour=F, linetype=F)




## Saline
sal.plot <- ggplot(sal.all, aes(x=BinTime, y=Mean, color=Season)) +
  geom_line(aes(linetype=Antibiotics), size=1) +
  geom_point(size=3, shape=21, aes(color=Season, fill=factor(ifelse(Antibiotics == "Yes", NA, Season)))) +
  scale_color_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3"), na.value="white", guide="none") +
  geom_errorbar(aes(ymin = Mean-SE, ymax = Mean+SE), width=0.125, position=position_dodge(0.1)) +
  ggtitle(expression("Saline"~~delta^13~"C")) +
  xlab("Time Since Gavage (hr)") +
  ylab(expression("Mean"~~delta^13~"C"~"(%"[o]~")")) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.title.x=element_text(size=14),
        axis.title.y=element_text(size=14),
        axis.text.x=element_text(size=12),
        axis.text.y=element_text(size=12),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank()) +
  guides(colour=F, linetype=F)




## Steal legend
steal.legend <- ggplot(sal.all, aes(x=BinTime, y=Mean, color=Season)) +
  geom_line(aes(linetype=Antibiotics), size=1) +
  geom_point(size=3) +
  scale_color_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  geom_errorbar(aes(ymin = Mean-SE, ymax = Mean+SE), width=0.125, position=position_dodge(0.1)) +
  ggtitle(expression("Saline"~~delta^13~"C")) +
  xlab("Time Since Gavage (hr)") +
  ylab(expression("Mean"~~delta^13~"C"~"(%"[o]~")")) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.title.x=element_text(size=14),
        axis.title.y=element_text(size=14),
        axis.text.x=element_text(size=12),
        axis.text.y=element_text(size=12),
        legend.title = element_text(size=20, face="bold"),
        legend.text = element_text(size=18)
        )
legend <- g_legend(steal.legend)






### All 3 substrates on same axis
carb.plot <- ggplot(crds.rem, aes(x=BinTime, y=Mean, color=Season)) +
  geom_line(aes(linetype=Antibiotics), size=1) +
   geom_point(size=3, shape=21, aes(color=Season, fill=factor(ifelse(Antibiotics == "Yes", NA, Season)))) +
  scale_color_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  scale_fill_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3"), na.value="white", guide="none") +
  geom_errorbar(aes(ymin = Mean-SE, ymax = Mean+SE), width=0.125, position=position_dodge(0.1)) + 
  facet_grid(.~Substrate) +
  xlab("Time Since Gavage (hr)") +
  ylab(expression("Mean"~~delta^13~"C"~"(%"[o]~")")) +
  xlim(-0.6,4.1) +
  scale_y_continuous(breaks=c(0, 500, 1000, 1500, 2000, 2500), limits=c(-30, 2700)) +
  theme(axis.text.x = element_text(size=16),
        axis.text.y = element_text(size=16),
        axis.title.x = element_text(size=20, margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y = element_text(size=20, margin=margin(t=0, r=10, b=, l=0)),
        strip.text.x = element_text(size=20, face="bold"),
        #strip.text.y = element_text(size=14),
        panel.spacing=unit(1, "lines"),
        legend.title = element_text(size = 20, face="bold"),
        legend.text = element_text(size=18),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.y = element_blank(),
        panel.grid.minor.x = element_blank(),
  )



tiff("crds.tiff", width = 12.75, height = 3.5, units = "in", res = 300)
carb.plot
dev.off()
```



### Inulin MetaG Endo/Exo Total Protein
```{r}
### Import total Endo / Exo / Protein Counts
endoexo <- read.csv("eCAMI/endo_exo_ORF_counts.csv", fileEncoding="UTF-8-BOM")
endoexo$Endo_RelAbund <- (endoexo$Endo / endoexo$Total_Protein)*100
endoexo$Exo_RelAbund <- (endoexo$Exo / endoexo$Total_Protein)*100
endoexo$Season <- c("Spring", "Spring", "Spring", "Summer", "Summer", "Summer", "Winter", "Winter", "Winter")
endoexo$Season <- ordered(endoexo$Season, levels=c("Summer", "Winter", "Spring"))

### Test Endo / Exo Rel Abund for Normality
hist(endoexo$Endo_RelAbund)
qqnorm(endoexo$Endo_RelAbund)
qqline(endoexo$Endo_RelAbund)
shapiro.test(endoexo$Endo_RelAbund)
  # p-value = 0.003571
  # Not Normal

hist(endoexo$Exo_RelAbund)
qqnorm(endoexo$Exo_RelAbund)
qqline(endoexo$Exo_RelAbund)
shapiro.test(endoexo$Exo_RelAbund)
  # p-value = 0.1865
  # Normal


### Test Endo
kruskal.test(formula = Endo_RelAbund ~ Season, data=endoexo)
  # p-value = 0.9481


### Test Exo
summary(aov(Exo_RelAbund ~ Season, data=endoexo))
  # p-value = 0.901


### Plot Endo
endo.plot <- ggplot(endoexo, aes(x=Season, y=Endo_RelAbund, fill=Season)) +
  geom_dotplot(binaxis='y', stackdir='center') +
  scale_fill_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Endo-Inulinase") +
  ylab("Relative Abundance (%)") +
  scale_y_continuous(breaks=c(0.00, 0.0125, 0.025, 0.0375, 0.05), limits=c(0, 0.05)) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.text.x = element_text(size=14),
        axis.text.y = element_text(size=14),
        axis.title.x = element_text(size=20, margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y = element_text(size=20, margin=margin(t=0, r=10, b=0, l=0)),
        strip.text.x = element_text(size=20),
        #strip.text.y = element_text(size=14),
        panel.spacing=unit(1, "lines"),
        legend.title = element_text(size = 18),
        legend.text = element_text(size=16),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
  )+
  guides(fill=F)

tiff("endo.mdtp.tiff", width = 5.5, height = 3.5, units = "in", res = 300)
endo.plot
dev.off()


### Plot Exo

legend <- g_legend(exo.plot)

exo.plot <- ggplot(endoexo, aes(x=Season, y=Exo_RelAbund, fill=Season)) +
  geom_dotplot(binaxis='y', stackdir='center') +
  scale_fill_manual(name="Season", values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  ggtitle("Exo-Inulinase") +
  ylab("Relative Abundance (%)") +
  scale_y_continuous(breaks=c(0.00, 0.0125, 0.025, 0.0375, 0.05), limits=c(0, 0.05)) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.text.x = element_text(size=14),
        axis.text.y = element_text(size=14),
        axis.title.x = element_text(size=20, margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y = element_text(size=20, margin=margin(t=0, r=10, b=0, l=0)),
        strip.text.x = element_text(size=20),
        strip.text.y = element_text(size=14),
        panel.spacing=unit(1, "lines"),
        legend.title = element_text(size = 18, face="bold"),
        legend.text = element_text(size=16),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
  ) +
  guides(fill=F)

tiff("exo.mdtp.tiff", width = 5.5, height = 3.5, units = "in", res = 300)
exo.plot
dev.off()



tiff("endo.exo.mdtp.tiff", width = 12.5, height = 3.5, units = "in", res = 300)
grid.newpage()
pushViewport(viewport(layout=(grid.layout(1,3, widths=c(1,1,0.25), heights=c(1,1,1)))))
print(endo.plot, vp=viewport(layout.pos.row=1, layout.pos.col=1))
print(exo.plot, vp=viewport(layout.pos.row=1, layout.pos.col=2))
legend$vp <- viewport(layout.pos.row=1, layout.pos.col=3)
grid.draw(legend)
dev.off()

```

### Inulinase Endo / Exo RPKG
```{r}
### Exo - All
exo_mapped_all <- read.csv('eCAMI/mapped_reads_all.csv')
exo_mapped_all <- exo_mapped_all[which(exo_mapped_all$Protein == "Exo"),]

shapiro.test(exo_mapped_all$RelAbund.100000)
  # p-value = 0.0005701

kruskal.test(formula = RelAbund.100000 ~ Season, data = exo_mapped_all)
  # p-value = 0.8342


### Exo - Separate
exo_mapped_sep <- read.csv('eCAMI/exoRPKG.csv')

shapiro.test(exo_mapped_sep$RPKG)
  # p-value = 3.348e-05

kruskal.test(formula = RPKG ~ Season, data = exo_mapped_sep)
  # p-value = 0.9322
```



### Inulin MetaG Exo Taxa Import
```{r}
####### PHYLUM #######

### Import Data
exo.shared <- read.csv("kaiju/exo/exo.phylum.csv")
rownames(exo.shared) <- exo.shared$X
exo.phylum <- data.frame(exo.shared$X)
rownames(exo.phylum) <- exo.phylum[,1]
exo.shared <- exo.shared[,-1]

### Make physeq object
exo.otu <- otu_table(exo.shared, taxa_are_rows=T)
meta.metag <- read.csv('metadata_metag.csv')
rownames(meta.metag) <- meta.metag$ID
meta.metag.samp <- sample_data(meta.metag)
exo.phylum <- tax_table(exo.phylum)
taxa_names(exo.phylum) <- tax_table(exo.phylum)
exo.physeq <- merge_phyloseq(exo.otu, meta.metag.samp, exo.phylum)

### Summarize data
exo.df <- psmelt(exo.physeq)
exo.sum <- exo.df %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))

### Save
#write.csv(exo.sum, "kaiju/exo.phylum.sum.csv", row.names=F)







####### CLASS #######

### Import Data
exo.shared <- read.csv("kaiju/exo/exo.class.csv")
rownames(exo.shared) <- exo.shared$X
exo.class <- data.frame(exo.shared$X)
rownames(exo.class) <- exo.class[,1]
exo.shared <- exo.shared[,-1]

### Make physeq object
exo.otu <- otu_table(exo.shared, taxa_are_rows=T)
meta.metag <- read.csv('metadata_metag.csv')
rownames(meta.metag) <- meta.metag$ID
meta.metag.samp <- sample_data(meta.metag)
exo.class <- tax_table(exo.class)
exo.physeq <- merge_phyloseq(exo.otu, meta.metag.samp, exo.class)

### Summarize data
exo.df <- psmelt(exo.physeq)
exo.sum <- exo.df %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))

### Save
#write.csv(exo.sum, "kaiju/exo.class.sum.csv", row.names=F)






####### ORDER #######

### Import Data
exo.shared <- read.csv("kaiju/exo/exo.order.csv")
rownames(exo.shared) <- exo.shared$X
exo.order <- data.frame(exo.shared$X)
rownames(exo.order) <- exo.order[,1]
exo.shared <- exo.shared[,-1]

### Make physeq object
exo.otu <- otu_table(exo.shared, taxa_are_rows=T)
meta.metag <- read.csv('metadata_metag.csv')
rownames(meta.metag) <- meta.metag$ID
meta.metag.samp <- sample_data(meta.metag)
exo.order <- tax_table(exo.order)
taxa_names(exo.order) <- exo.order
exo.physeq <- merge_phyloseq(exo.otu, meta.metag.samp, exo.order)

### Summarize data
exo.df <- psmelt(exo.physeq)
exo.sum <- exo.df %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))

### Save
#write.csv(exo.sum, "kaiju/exo.order.sum.csv", row.names=F)






####### FAMILY #######

### Import Data
exo.shared <- read.csv("kaiju/exo/exo.family.csv")
rownames(exo.shared) <- exo.shared$X
exo.family <- data.frame(exo.shared$X)
rownames(exo.family) <- exo.family[,1]
exo.shared <- exo.shared[,-1]

### Make physeq object
exo.otu <- otu_table(exo.shared, taxa_are_rows=T)
meta.metag <- read.csv('metadata_metag.csv')
rownames(meta.metag) <- meta.metag$ID
meta.metag.samp <- sample_data(meta.metag)
exo.family <- tax_table(exo.family)
taxa_names(exo.family) <- exo.family
exo.physeq <- merge_phyloseq(exo.otu, meta.metag.samp, exo.family)

### Summarize data
exo.df <- psmelt(exo.physeq)
exo.sum <- exo.df %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))

### Save
#write.csv(exo.sum, "kaiju/exo.family.sum.csv", row.names=F)






####### GENUS #######

### Import Data
exo.shared <- read.csv("kaiju/exo/exo.genus.csv")
rownames(exo.shared) <- exo.shared$X
exo.genus <- data.frame(exo.shared$X)
rownames(exo.genus) <- exo.genus[,1]
exo.shared <- exo.shared[,-1]

### Make physeq object
exo.otu <- otu_table(exo.shared, taxa_are_rows=T)
meta.metag <- read.csv('metadata_metag.csv')
rownames(meta.metag) <- meta.metag$ID
meta.metag.samp <- sample_data(meta.metag)
exo.genus <- tax_table(exo.genus)
taxa_names(exo.genus) <- exo.genus
exo.physeq <- merge_phyloseq(exo.otu, meta.metag.samp, exo.genus)

### Summarize data
exo.df <- psmelt(exo.physeq)
exo.sum <- exo.df %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))

### Save
#write.csv(exo.sum, "kaiju/exo.genus.sum.csv", row.names=F)







####### SPECIES #######

### Import Data
exo.shared <- read.csv("kaiju/exo/exo.species.csv")
rownames(exo.shared) <- exo.shared$X
exo.species <- data.frame(exo.shared$X)
rownames(exo.species) <- exo.species[,1]
exo.shared <- exo.shared[,-1]

### Make physeq object
exo.otu <- otu_table(exo.shared, taxa_are_rows=T)
meta.metag <- read.csv('metadata_metag.csv')
rownames(meta.metag) <- meta.metag$ID
meta.metag.samp <- sample_data(meta.metag)
exo.species <- tax_table(exo.species)
taxa_names(exo.species) <- exo.species
exo.physeq <- merge_phyloseq(exo.otu, meta.metag.samp, exo.species)

### Summarize data
exo.df <- psmelt(exo.physeq)
exo.sum <- exo.df %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))

### Save
#write.csv(exo.sum, "kaiju/exo.species.sum.csv", row.names=F)
```

### Inulin MetaG Exo Rel Abund
```{r}
# Calculate rel abund
exo.ra <- transform_sample_counts(exo.gen, function(x){(x/sum(x))*100})

# Create rank abundance plot
barplot(sort(taxa_sums(exo.ra),TRUE)[1:20]/nsamples(exo.ra),las=2,cex.axis=.7)

# Pull out top 10 genera
top10 <- as.data.frame(sort(taxa_sums(exo.ra),TRUE)[1:10]/nsamples(exo.ra))
exo.df <- psmelt(exo.ra)

exo.top <- exo.df[which(exo.df$OTU == rownames(top10)[1]),]
exo.top[10:18,] <- exo.df[which(exo.df$OTU == rownames(top10)[3]),]
exo.top[19:27,] <- exo.df[which(exo.df$OTU == rownames(top10)[5]),]
exo.top[28:36,] <- exo.df[which(exo.df$OTU == rownames(top10)[6]),]
exo.top[37:45,] <- exo.df[which(exo.df$OTU == rownames(top10)[7]),]
exo.top[46:54,] <- exo.df[which(exo.df$OTU == rownames(top10)[9]),]
exo.top$Genus <- ordered(exo.top$Genus, levels = c("Bacteroides", "Muribaculum", "Cohnella", "Prevotella", "Parabacteroides", "Asaia"))

# exo.top$Genus <- factor(c(rep("Bacteroides",9), rep("Unclassified",9), rep("Muribaculum",9),
#                           rep("Bacteria - Unclassified",9), rep("Cohnella", 9), rep("Prevotella", 9),
#                           rep("Parabacteroides",9), rep("Muribaculaceae - Unclassified",9),
#                           rep("Asaia", 9), rep("Clostridiales - Unclassified", 9)))


#Summarize data
exo.sum <- exo.top %>%
  group_by(Genus) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))

### Plot
exo.gen.mdtp <- ggplot(exo.top, aes(x=Genus, y=Abundance, fill=Genus)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=3, stackratio=1.15) +
  facet_grid(Season~.) +
  ylab("Relative Abundance (%)") +
  ggtitle("Top 6 Genera") +
  scale_fill_manual(values=c("red2", "gold", "deepskyblue1","darkorchid1", "papayawhip", "gray12")) +
  ylim(0,30) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.title.x=element_text(size=18, 
                                  margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y=element_text(size=18),
        axis.text.x=element_text(size=14, angle=25, hjust=1),
        axis.text.y=element_text(size=14),
        plot.margin = unit(c(0, 0.3, 0.3, 0.3), "cm"),
          #top, right, bottom, left
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.text.y = element_text(size=14, face="bold"),
        strip.background = element_blank(),
        legend.title = element_text(size=15, face="bold"),
        legend.text=element_text(size=15),
        panel.spacing.y=unit(1,"lines")
  )

tiff("exo.genus.mdtp.tiff", width=8.45, height=6.2, units="in", res=600)
exo.gen.mdtp
dev.off()


### Make plot w/ top 6 genera + 2 unclassified
exo.top.unclass <- exo.df[which(exo.df$OTU == rownames(top10)[1]),]
exo.top.unclass[10:18,] <- exo.df[which(exo.df$OTU == rownames(top10)[2]),]
exo.top.unclass[19:27,] <- exo.df[which(exo.df$OTU == rownames(top10)[3]),]
exo.top.unclass[28:36,] <- exo.df[which(exo.df$OTU == rownames(top10)[4]),]
exo.top.unclass[37:45,] <- exo.df[which(exo.df$OTU == rownames(top10)[5]),]
exo.top.unclass[46:54,] <- exo.df[which(exo.df$OTU == rownames(top10)[6]),]
exo.top.unclass[55:63,] <- exo.df[which(exo.df$OTU == rownames(top10)[7]),]
exo.top.unclass[64:72,] <- exo.df[which(exo.df$OTU == rownames(top10)[9]),]
exo.top.unclass$Genus <- ordered(exo.top.unclass$Genus, levels = c("Bacteroides", "Muribaculum", "Cohnella", "Prevotella", "Parabacteroides", "Asaia", "Unclassified Domain", "Unclassified Bacteria"))
exo.top.unclass$Genus[10:18] <- "Unclassified Domain"
exo.top.unclass$Genus[28:36] <- "Unclassified Bacteria"



#Summarize data
exo.unclass.sum <- exo.top.unclass %>%
  group_by(Genus) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))

### Plot
exo.gen.unclass.mdtp <- ggplot(exo.top.unclass, aes(x=Genus, y=Abundance, fill=Genus)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=3, stackratio=1.15) +
  facet_grid(Season~.) +
  ylab("Relative Abundance (%)") +
  ggtitle("Top 6 Genera + Unclassified") +
  scale_fill_manual(values=c("red2", "gold", "deepskyblue1","darkorchid1", "papayawhip", "gray12", "gray47", "gray86")) +
  ylim(0,30) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.title.x=element_text(size=18, 
                                  margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y=element_text(size=18),
        axis.text.x=element_text(size=14, angle=25, hjust=1),
        axis.text.y=element_text(size=14),
        plot.margin = unit(c(0, 0.3, 0.3, 0.3), "cm"),
          #top, right, bottom, left
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.text.y = element_text(size=14, face="bold"),
        strip.background = element_blank(),
        legend.title = element_text(size=15, face="bold"),
        legend.text=element_text(size=15),
        panel.spacing.y=unit(1,"lines")
  )

tiff("exo.genus.unclass.mdtp.tiff", width=10.55, height=6.2, units="in", res=600)
exo.gen.unclass.mdtp
dev.off()

```

### Inulin MetaG Exo NMDS
```{r}
### Bray-Curtis NMDS
exo.nmds <- ordinate(physeq=exo.physeq, method="NMDS", distance="bray")
  # stress = 0.1096339

tiff("exo.nmds.mdtp.tiff", width=8, height=6, units="in", res=600)
plot_ordination(physeq=exo.physeq, ordination=exo.nmds, color="Season") + 
  scale_color_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3")) +
  geom_path(data=exo.physeq.elip.df, aes(x=NMDS1, y=NMDS2, colour=group), size=1, linetype=1) +
  ggtitle("Exo-Inulinase") +
  geom_point(size=3) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.text.x = element_text(size=20),
        axis.text.y = element_text(size=20),
        axis.title.x = element_text(size=20),
        axis.title.y = element_text(size=20),
        legend.title = element_text(size=20, face="bold"),
        legend.text = element_text(size=20),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
        )
dev.off()


### NMDS Ellipses
sample_data(exo.physeq)$Elip <- paste(sample_data(exo.physeq)$Season)
exo.physeq.elip <- data.frame(MDS1 = exo.nmds$points[,1], MDS2 = exo.nmds$points[,2],
                           Group = sample_data(exo.physeq)$Season)
plot.new()
exo.physeq.elip.ord <- ordiellipse(exo.nmds, sample_data(exo.physeq)$Elip, display="sites", kind="se", conf=0.95, label=T)
exo.physeq.elip.df <- data.frame()
for(i in levels(exo.physeq.elip$Group)){
  exo.physeq.elip.df <- rbind(exo.physeq.elip.df,
                           cbind(as.data.frame(with(exo.physeq.elip[exo.physeq.elip$Group==i,],
                                                    veganCovEllipse(exo.physeq.elip.ord[[i]]$cov,
                                                                    exo.physeq.elip.ord[[i]]$center,
                                                                    exo.physeq.elip.ord[[i]]$scale))), group=i
                                 ))
}

### Bray-Curtis PCoA
exo.pcoa <- ordinate(physeq=exo.physeq, method="PCoA", distance="bray")

plot_ordination(physeq=exo.physeq, ordination=exo.pcoa, color="Season") + 
  scale_color_manual(values=c("darkgoldenrod2", "dodgerblue3", "green3"))



### PERMANOVA for NMDS
exo.physeq.sampdf <- data.frame(sample_data(exo.physeq))
exo.bray<- phyloseq::distance(physeq=exo.physeq, method="bray")

## Season
adonis(exo.bray~ Season, data=exo.physeq.sampdf)
  # p-value = 0.902
beta.tax = betadisper(d=exo.bray, group=exo.physeq.sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.604




# ### Import data
# P01 <- read.csv("eCAMI/P01_counts.csv", fileEncoding="UTF-8-BOM")
# P07 <- read.csv("eCAMI/P07_counts.csv", fileEncoding="UTF-8-BOM")
# P11 <- read.csv("eCAMI/P11_counts.csv", fileEncoding="UTF-8-BOM")
# S07 <- read.csv("eCAMI/S07_counts.csv", fileEncoding="UTF-8-BOM")
# S29 <- read.csv("eCAMI/S29_counts.csv", fileEncoding="UTF-8-BOM")
# S34 <- read.csv("eCAMI/S34_counts.csv", fileEncoding="UTF-8-BOM")
# W31 <- read.csv("eCAMI/W31_counts.csv", fileEncoding="UTF-8-BOM")
# W32 <- read.csv("eCAMI/W32_counts.csv", fileEncoding="UTF-8-BOM")
# W33 <- read.csv("eCAMI/W33_counts.csv", fileEncoding="UTF-8-BOM")
# 
# ### Remove endo rows
# P07 <- P07[-1,]
# S07 <- S07[-1,]
# S34 <- S34[-1,]
# W33 <- W33[-1,]



```





### Gene Length
```{r}
geneLength <- read.csv('eCAMI/geneLength.csv')
endoLength <- geneLength[which(geneLength$Gene == "Endo"),]
exoLength <- geneLength[which(geneLength$Gene == "Exo"),]

shapiro.test(endoLength$Length..bp.)
  # p-value = 0.0005404
shapiro.test(exoLength$Length..bp.)
  # p-value < 2.2e-16

kruskal.test(formula = Length..bp. ~ Season, data = exoLength)
  # p-value = 0.14
kruskal.test(formula = Length..bp. ~ Substrate, data = exoLength)
  # p-value = 0.9816
summary(aov(Length..bp. ~ Season + Substrate, data = exoLength))
  # Season = 0.488
  # Substrate = 0.862
```
### Average genome size /  Genome Equivalents
```{r}
ags <- read.csv('AverageGenomeSize.csv')

shapiro.test(ags$Avg_Genome_Size)
  # p-value = 0.4778
shapiro.test(ags$Genome_Equivalents)
  # p-value = 0.1762


summary(aov(Avg_Genome_Size ~ Season, data = ags))
  # p-value = 0.298
summary(aov(Genome_Equivalents ~ Season, data = ags))
  # p-value = 0.254
```
### Inulinase 