---
title: "Code_16S"
author: "Edna Chiang"
date: "2/24/2022"
output: html_document
---

### Load Libraries
```{r}
library(ape)
library(dplyr)
library(dunn.test)
library(effsize)
library(ggplot2)
library(grid)
library(gridExtra)
library(phyloseq)
library(picante)
library(pwr)
library(tidyr)
library(vegan)
library(miLineage)
library(equivalence)
source('scripts/misc.R')
source('scripts/simper_pretty.R')
source('scripts/R_krusk.R')
source('scripts/removeTaxa.R')
theme_set(theme_bw())
set.seed(1)
```

### DON'T RUN - Import Data
```{r}
# Link files
shared = "16S/final.0.03.norm.shared"
taxonomy = "16S/final.taxonomy"

# Import mothur data
mothurdata = import_mothur(
  mothur_shared_file = shared,
  mothur_constaxonomy_file = taxonomy)

# Add OTU column in taxonomy table
tax_table(mothurdata) <- cbind(tax_table(mothurdata), 
                               row.names(tax_table(mothurdata)))

# Rename taxonomy table columns
colnames(tax_table(mothurdata)) <- c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")

### Prepare metadata
meta <- read.csv("metadata.csv")
rownames(meta) <- meta$Sample
divsum <- read.table("16S/final.0.03.norm.groups.summary", header = T)
divsum$test <- strsplit(divsum$group, "_")
dna <- read.csv("dna.csv")

# Fill in metadata
for(i in 1:nrow(divsum)){
  divsum$Individual[i] <- divsum$test[[i]][[2]]

  use <- which(meta$ID == divsum$Individual[i])
  divsum$Source[i] <- divsum$test[[i]][[1]]
  divsum[i,18:30] <- meta[use,]

  dna.use <- which(dna$Sample == divsum$group[i])
  divsum$DNAyield[i] <- dna$DNA_Yield_.ng.mg.[dna.use]
}

divsum <- divsum[,-1]
divsum <- divsum[,-14:-15]
colnames(divsum)[1] <- "Sample"
divsum$Source <- substr(divsum$Source, start = 2, stop = 2)
divsum$Source[which(divsum$Source == "C")] <- "Content"
divsum$Source[which(divsum$Source == "T")] <- "Mucosa"




# ### Prepare CRDS metrics
# load('crds.norm.RData')
# load('crds.use.Rdata')
# crds.meta <- read.csv('metadata.csv')
# crds.use[, colnames(crds.meta)[3:ncol(crds.meta)]] <- NA
# for(i in 1:nrow(crds.use)){
#   crds.meta.row <- which(crds.meta$ID == crds.use$ID[i])
#   crds.use[i, 7:ncol(crds.use)] <- crds.meta[crds.meta.row, 3:ncol(crds.meta)]
# }
# 
# # Calculate square root of AUC
# crds.use$AUC <- as.numeric(crds.use$AUC)
# minAUC <- min(crds.use$AUC)
# crds.use$AUC_sqrt <- crds.use$AUC + (minAUC * -1)
# crds.use$AUC_sqrt <- sqrt(crds.use$AUC_sqrt)
# 
# # Remove outliers
# crds.rem <- crds.use[-which(crds.use$ID == "S23"),]
# crds.rem <- crds.rem[-which(crds.rem$ID == "S37"),]
# crds.rem <- crds.rem[-which(crds.rem$ID == "S60"),]
# crds.rem <- crds.rem[-which(crds.rem$ID == "W43"),]
# crds.rem <- crds.rem[-which(crds.rem$ID == "W20"),]
#   # This is due to odd crds.metagenome
# crds.rem <- crds.rem[-which(crds.rem$ID == "W50"),]
# 
# # Remove samples w/o CRDS metrics
# #divsum <- divsum[-which(divsum$group == "CC1_S11"),]
# #divsum <- divsum[-which(divsum$group == "CC1_S60"),]
# #divsum <- divsum[-which(divsum$group == "CC2_S37"),]
#   # Since miLineage was not sig, no reason to remove these samples
# 
# 
# 
# # Fill in metadata
# for(i in 1:nrow(divsum)){
#   divsum$Individual[i] <- divsum$test[[i]][[2]]
#   
#   use <- which(meta$ID == divsum$Individual[i])
#   divsum$Source[i] <- divsum$test[[i]][[1]]
#   divsum[i,18:30] <- meta[use,]
#   
#   crdsRow <- which(crds.rem$ID == divsum$Individual[i])
#   divsum$Baseline[i] <- crds.rem$Baseline[crdsRow]
#   divsum$AUC_sqrt[i] <- crds.rem$AUC_sqrt[crdsRow]
#   divsum$AvgD[i] <- crds.rem$AvgD[crdsRow]
#   divsum$MaxD[i] <- crds.rem$MaxD[crdsRow]
#   divsum$MaxSlope[i] <- crds.rem$MaxSlope[crdsRow]
#   
#   dna.use <- which(dna$Sample == divsum$group[i])
#   divsum$DNAyield[i] <- dna$DNA_Yield_.ng.mg.[dna.use]
# }
# 
# divsum <- divsum[,-1]
# divsum <- divsum[,-14:-15]
# colnames(divsum)[1] <- "Sample"
# divsum$Source <- substr(divsum$Source, start = 2, stop = 2)
# divsum$Source[which(divsum$Source == "C")] <- "Content"
# divsum$Source[which(divsum$Source == "T")] <- "Mucosa"



### Add metadata to phyloseq object
meta.ready <- sample_data(divsum)
sample_names(meta.ready) <- divsum$Sample
physeq <- merge_phyloseq(mothurdata, meta.ready)


# Remove OTUs that appear in < 3 samples
otu <- data.frame(otu_table(physeq))
otu.rem <- list()
for(j in 1:nrow(otu)){
  empty <- length(which(otu[j,] == 0))
    # Identifies # of samples w/o OTU
  if(empty > 42){
    # If OTU appears in < 3 samples
    otu.rem <- append(otu.rem, j)
      # Create list of OTUs that appear in only 1 sample
  }
}
# 1251 OTUs to remove
otu.rem <- otu[-unlist(otu.rem),]


# Update phyloseq object w/ removed OTUs
phy.rem <- merge_phyloseq(otu_table(otu.rem, taxa_are_rows=T), sample_data(physeq), tax_table(physeq))
 

# Order factors
sample_data(phy.rem)$Season <- ordered(sample_data(phy.rem)$Season,
                                     levels=c("Summer", "Winter", "Spring"))
sample_data(phy.rem)$Antibiotics <- ordered(sample_data(phy.rem)$Antibiotics,
                                     levels=c("No", "Yes"))
sample_data(phy.rem)$Sex <- ordered(sample_data(phy.rem)$Sex,
                                     levels=c("Female", "Male"))
sample_data(phy.rem)$Substrate <- ordered(sample_data(phy.rem)$Substrate,
                                     levels=c("Inulin", "Saline"))


# Calculate relative abundance
phy.ra <- transform_sample_counts(phy.rem, function(x){(x/sum(x))*100})

# Separate by Sample Type
phy.c <- subset_samples(phy.ra, Source == "Content")
phy.m <- subset_samples(phy.ra, Source == "Mucosa")

# Separate by Season
phy.sum <- subset_samples(phy.ra, Season == "Summer")
phy.win <- subset_samples(phy.ra, Season == "Winter")
phy.spr <- subset_samples(phy.ra, Season == "Spring")

# Separate by Season + Sample Type
phy.sum.c <- subset_samples(phy.sum, Source == "Content")
  # 17 samples
phy.sum.m <- subset_samples(phy.sum, Source == "Mucosa")
  # 15 samples
phy.win.c <- subset_samples(phy.win, Source == "Content")
  # 5 samples
phy.win.m <- subset_samples(phy.win, Source == "Mucosa")
  # 1 sample
phy.spr.c <- subset_samples(phy.spr, Source == "Content")
  # 4 samples
phy.spr.m <- subset_samples(phy.spr, Source == "Mucosa")
  # 2 samples


# Save phyloseq objects
#save(phy.rem, file = "16S/physeq/phy.all.counts.RData")
#save(phy.ra, file = "16S/physeq/phy.all.relabund.RData")
#save(phy.c, file = "16S/physeq/phy.content.RData")
#save(phy.m, file = "16S/physeq/phy.mucosa.RData")
#save(phy.sum.c, file = "16S/physeq/phy.summer.content.RData")
#save(phy.sum.m, file = "16S/physeq/phy.summer.mucosa.RData")
#save(phy.win.c, file = "16S/physeq/phy.winter.content.RData")
#save(phy.spr.c, file = "16S/physeq/phy.spring.content.RData")
```



### Load Data
```{r}
load("16S/physeq/phy.all.counts.RData")
load("16S/physeq/phy.all.relabund.RData")
load("16S/physeq/phy.content.RData")
load("16S/physeq/phy.mucosa.RData")
load("16S/physeq/phy.summer.content.RData")
load("16S/physeq/phy.summer.mucosa.RData")
load("16S/physeq/phy.winter.content.RData")
load("16S/physeq/phy.spring.content.RData")
load('crds.use.Rdata')

meta <- data.frame(sample_data(phy.rem))

```


### miLineage - OTUs associated w/ Delta metrics
```{r}
### Format phyloseq objects
phy.con.raw <- subset_samples(phy.rem, Source == "Content")
phy.c.rem <- filter_taxa(phy.con.raw, function(x) sum(x) > 0, T)
phy.muc.raw <- subset_samples(phy.rem, Source == "Mucosa")
phy.m.rem <- filter_taxa(phy.muc.raw, function(x) sum(x) > 0, T)


### Format tax table
tax.con <- data.frame(tax_table(phy.c.rem))
colnames(tax.con) <- c("Rank1", "Rank2", "Rank3", "Rank4", "Rank5", "Rank6", "Rank7")
tax.con <- as.matrix(tax.con)
tax.muc <- data.frame(tax_table(phy.m.rem))
colnames(tax.muc) <- c("Rank1", "Rank2", "Rank3", "Rank4", "Rank5", "Rank6", "Rank7")
tax.muc <- as.matrix(tax.muc)                    
                    

### Format metadata
meta.con <- meta[which(meta$Source == "Content"),]
meta.muc <- meta[which(meta$Source == "Mucosa"),]
delta.c <- meta.con[,24:33]
delta.c$BodyMass <- as.numeric(meta.con$BodyMass)
delta.c$DNAyield <- meta.con$DNAyield
delta.c$Baseline <- as.numeric(delta.c$Baseline)
delta.c$AUC_sqrt <- as.numeric(delta.c$AUC_sqrt)
delta.c$AvgD <- as.numeric(delta.c$AvgD)
delta.c$MaxD <- as.numeric(delta.c$MaxD)
delta.c$MaxSlope <- as.numeric(delta.c$MaxSlope)
delta.c$BodyMass <- as.numeric(delta.c$BodyMass)
delta.c$DNAyield <- as.numeric(delta.c$DNAyield)
delta.c <- as.matrix(delta.c)

delta.m <- meta.muc[,24:33]
delta.m$BodyMass <- meta.muc$BodyMass
delta.m$DNAyield <- meta.muc$DNAyield
delta.m$BodyMass <- as.numeric(meta.muc$BodyMass)
delta.m$DNAyield <- meta.muc$DNAyield
delta.m$Baseline <- as.numeric(delta.m$Baseline)
delta.m$AUC_sqrt <- as.numeric(delta.m$AUC_sqrt)
delta.m$AvgD <- as.numeric(delta.m$AvgD)
delta.m$MaxD <- as.numeric(delta.m$MaxD)
delta.m$MaxSlope <- as.numeric(delta.m$MaxSlope)
delta.m$BodyMass <- as.numeric(delta.m$BodyMass)
delta.m$DNAyield <- as.numeric(delta.m$DNAyield)
delta.m <- as.matrix(delta.m)


delta.use <- delta.c[,-2:-3]
delta.use <- delta.use[,-3]
delta.use <- delta.use[,-3]


### Format OTU table
otu.c <- as.matrix(t(data.frame(otu_table(phy.c.rem))))
otu.m <- as.matrix(t(data.frame(otu_table(phy.m.rem))))





all.phy.c <- QCAT_GEE(OTU=otu.c, Tax=tax.con, X=delta.use, X.index = 1, Z=delta.use, Z.index=1,  fdr.alpha=0.05, n.resample = 99)
  # No sig
#save(all.phy.c, file = "miLineage.output.all.phy.c.RData")



all.phy.m <- QCAT_GEE(OTU=otu.m, Tax=tax.muc, X=delta.use, X.index = 2, Z=delta.use, Z.index=2,  fdr.alpha=0.05, n.resample = 999)
  # No sig
#save(all.phy.m, file = "miLineage.output.all.phy.m.RData")





### Format phyloseq objects
phy.con.raw <- subset_samples(phy.rem, Source == "Content")
phy.con.sum <- subset_samples(phy.con.raw, Season == "Summer")
phy.con.sum <- filter_taxa(phy.con.sum, function(x) sum(x) > 0, T)


### Format tax table
tax.con <- data.frame(tax_table(phy.con.sum))
colnames(tax.con) <- c("Rank1", "Rank2", "Rank3", "Rank4", "Rank5", "Rank6", "Rank7")
tax.con <- as.matrix(tax.con)


### Format metadata
meta.con <- data.frame(sample_data(phy.con.sum))
delta.con.sum <- meta.con[,28:33]
delta.con.sum$BodyMass <- meta.con$BodyMass
delta.con.sum$Baseline <- as.numeric(delta.con.sum$Baseline)
delta.con.sum$AUC_sqrt <- as.numeric(delta.con.sum$AUC_sqrt)
delta.con.sum$AvgD <- as.numeric(delta.con.sum$AvgD)
delta.con.sum$MaxD <- as.numeric(delta.con.sum$MaxD)
delta.con.sum$MaxSlope <- as.numeric(delta.con.sum$MaxSlope)
delta.con.sum$DNAyield <- as.numeric(delta.con.sum$DNAyield)
delta.con.sum$BodyMass <- as.numeric(delta.con.sum$BodyMass)
delta.con.sum <- as.matrix(delta.con.sum)

delta.use <- delta.con.sum[,-2:-3]
delta.use <- delta.use[,-3]


### Format OTU table
otu.c <- as.matrix(t(data.frame(otu_table(phy.con.sum))))

phy.con.sum.qcat <- QCAT_GEE(OTU=otu.c, Tax=tax.con, X=delta.use, X.index = 2,
                              Z=delta.con.sum, Z.index=2, fdr.alpha=0.05, n.resample = 999)
#save(phy.con.sum.qcat, file = "miLineage.output.phy.con.sum.RData")



### Format phyloseq objects
phy.con.noabx.raw <- subset_samples(phy.con.sum, Antibiotics == "No")
phy.con.noabx.raw <- filter_taxa(phy.con.noabx.raw, function(x) sum(x) > 0, T)


### Format tax table
tax.con <- data.frame(tax_table(phy.con.noabx.raw))
colnames(tax.con) <- c("Rank1", "Rank2", "Rank3", "Rank4", "Rank5", "Rank6", "Rank7")
tax.con <- as.matrix(tax.con)


### Format metadata
meta.con <- data.frame(sample_data(phy.con.noabx.raw))
delta.con.noabx <- meta.con[,28:33]
delta.con.noabx$BodyMass <- meta.con$BodyMass
delta.con.noabx$Baseline <- as.numeric(delta.con.noabx$Baseline)
delta.con.noabx$AUC_sqrt <- as.numeric(delta.con.noabx$AUC_sqrt)
delta.con.noabx$AvgD <- as.numeric(delta.con.noabx$AvgD)
delta.con.noabx$MaxD <- as.numeric(delta.con.noabx$MaxD)
delta.con.noabx$MaxSlope <- as.numeric(delta.con.noabx$MaxSlope)
delta.con.noabx$DNAyield <- as.numeric(delta.con.noabx$DNAyield)
delta.con.noabx$BodyMass <- as.numeric(delta.con.noabx$BodyMass)
#test <- delta.con.noabx[,-7]
delta.con.noabx <- as.matrix(delta.con.noabx)

delta.use <- delta.con.noabx[,-2:-3]
delta.use <- delta.use[,-3]



### Format OTU table
otu.c <- as.matrix(t(data.frame(otu_table(phy.con.noabx.raw))))

phy.con.noabx.qcat <- QCAT_GEE(OTU=otu.c, Tax=tax.con, X=delta.use, X.index = 2, Z=delta.use, Z.index=2,  fdr.alpha=0.05, n.resample = 99)
#save(phy.con.noabx.qcat, file = "miLineage.output.phy.con.noabx.RData")

```



### ABX-surviving microbes - Prepare data
```{r}
### Separate ABX/No ABX
phy.sum.c.noabx <- subset_samples(phy.sum.c, Antibiotics == "No")
  # 14 samples
phy.sum.c.abx <- subset_samples(phy.sum.c, Antibiotics == "Yes")
  # 3 samples
phy.sum.m.noabx <- subset_samples(phy.sum.m, Antibiotics == "No")
  # 12 samples
phy.sum.m.abx <- subset_samples(phy.sum.m, Antibiotics == "Yes")
  # 3 samples
phy.win.c.abx <- subset_samples(phy.win.c, Antibiotics == "Yes")
  # 4 samples
phy.spr.c.noabx <- subset_samples(phy.spr.c, Antibiotics == "No")
  # 4 samples


### Drop taxa in < 2 samples
otu.sum.c.abx.otu <- data.frame(otu_table(phy.sum.c.abx))
otu.sum.c.abx.otu.rem <- list()
for(a in 1:nrow(otu.sum.c.abx.otu)){
  empty <- length(which(otu.sum.c.abx.otu[a,] == 0))
    # Identifies # of samples w/o OTU
  if(empty == 3){
    # If OTU appears in 0 samples
    otu.sum.c.abx.otu.rem <- append(otu.sum.c.abx.otu.rem, a)
      # Create list of empty OTUs
  }
}
# 1655 OTUs to remove
otu.sum.c.abx.otu.rem <- otu.sum.c.abx.otu[-unlist(otu.sum.c.abx.otu.rem),]
# Update physeq object
phy.sum.c.abx.rem <- merge_phyloseq(otu_table(otu.sum.c.abx.otu.rem, taxa_are_rows=T),
                                    sample_data(phy.sum.c.abx), tax_table(phy.sum.c.abx))


### Drop taxa in 0 samples
otu.sum.m.abx.otu <- data.frame(otu_table(phy.sum.m.abx))
otu.sum.m.abx.otu.rem <- list()
for(b in 1:nrow(otu.sum.m.abx.otu)){
  empty <- length(which(otu.sum.m.abx.otu[b,] == 0))
    # Identifies # of samples w/o OTU
  if(empty == 3){
    # If OTU appears in 0 samples
    otu.sum.m.abx.otu.rem <- append(otu.sum.m.abx.otu.rem, b)
      # Create list of OTUs that appear in 0 sample
  }
}
# 1132 OTUs to remove
otu.sum.m.abx.otu.rem <- otu.sum.m.abx.otu[-unlist(otu.sum.m.abx.otu.rem),]
# Update physeq object
phy.sum.m.abx.rem <- merge_phyloseq(otu_table(otu.sum.m.abx.otu.rem, taxa_are_rows=T),
                                    sample_data(phy.sum.m.abx), tax_table(phy.sum.m.abx))



### Drop taxa in 0 samples
otu.win.c.abx.otu <- data.frame(otu_table(phy.win.c.abx))
otu.win.c.abx.otu.rem <- list()
for(c in 1:nrow(otu.win.c.abx.otu)){
  empty <- length(which(otu.win.c.abx.otu[c,] == 0))
    # Identifies # of samples w/o OTU
  if(empty == 4){
    # If OTU appears in 0 samples
    otu.win.c.abx.otu.rem <- append(otu.win.c.abx.otu.rem, c)
      # Create list of OTUs that appear in 0 sample
  }
}
# 2118 OTUs to remove
otu.win.c.abx.otu.rem <- otu.win.c.abx.otu[-unlist(otu.win.c.abx.otu.rem),]
# Update physeq object
phy.win.c.abx.rem <- merge_phyloseq(otu_table(otu.win.c.abx.otu.rem, taxa_are_rows=T),
                                    sample_data(phy.win.c.abx), tax_table(phy.win.c.abx))

```


### Test ABX S vs. W
```{r}
# Create physeq object
abx.use <- merge_phyloseq(phy.sum.c.abx.rem, phy.win.c.abx.rem)


##### Bray-Curtis tests
bray.abx <- phyloseq::distance(physeq = abx.use, method = "bray")
beta.tax <- betadisper(d = bray.abx, group = data.frame(sample_data(abx.use))$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.654
adonis(bray.abx ~ Season, data=data.frame(sample_data(abx.use)), permutations = 9999)
  # p-value = 0.1429


##### SIMPER
simper.pretty(data.frame(otu_table(abx.use)),
              data.frame(sample_data(abx.use)),
              c("Season"),
              perc_cutoff=0.5, low_cutoff='y', low_val=0.01,
              output_name='abx')
group.simper <- read.csv("abx_clean_simper.csv")
kruskal.pretty(data.frame(otu_table(abx.use)),
               data.frame(sample_data(abx.use)),
               csv = group.simper,
               c("Season"),
               output_name = 'abx',
               taxonomy = data.frame(tax_table(abx.use)))







##### TAXA WILCOXON
meta.abx <- data.frame(sample_data(abx.use))


# Phylum
abx.phy <- tax_glom(abx.use, taxrank="Phylum")
phy <- data.frame(otu_table(abx.phy))
wil.phy <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(abx.phy)))[1]){
  es <- cohen.d(as.numeric(phy[taxon,]) ~ meta.abx$Season)
  pwr <- pwr.t2n.test(n1 = length(which(meta.abx$Season == "Summer")),
            n2 = length(which(meta.abx$Season == "Winter")), d = es$estimate)
  wil.phy <- rbind(wil.phy, data.frame(taxon = rownames(phy)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.phy$fdr <- mt(abx.phy, classlabel = "Season")$fdr



# Class
abx.cla <- tax_glom(abx.use, taxrank="Class")
cla <- data.frame(otu_table(abx.cla))
wil.cla <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(abx.cla)))[1]){
  es <- cohen.d(as.numeric(cla[taxon,]) ~ meta.abx$Season)
  pwr <- pwr.t2n.test(n1 = length(which(meta.abx$Season == "Summer")),
            n2 = length(which(meta.abx$Season == "Winter")), d = es$estimate)
  wil.cla <- rbind(wil.cla, data.frame(taxon = rownames(cla)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.cla$fdr <- mt(abx.cla, classlabel = "Season")$fdr



# Order
abx.ord <- tax_glom(abx.use, taxrank="Order")
ord <- data.frame(otu_table(abx.ord))
wil.ord <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(abx.ord)))[1]){
  es <- cohen.d(as.numeric(ord[taxon,]) ~ meta.abx$Season)
  pwr <- pwr.t2n.test(n1 = length(which(meta.abx$Season == "Summer")),
            n2 = length(which(meta.abx$Season == "Winter")), d = es$estimate)
  wil.ord <- rbind(wil.ord, data.frame(taxon = rownames(ord)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.ord$fdr <- mt(abx.ord, classlabel = "Season")$fdr




# Family
abx.fam <- tax_glom(abx.use, taxrank="Family")
fam <- data.frame(otu_table(abx.fam))
wil.fam <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(abx.fam)))[1]){
  es <- cohen.d(as.numeric(fam[taxon,]) ~ meta.abx$Season)
  pwr <- pwr.t2n.test(n1 = length(which(meta.abx$Season == "Summer")),
            n2 = length(which(meta.abx$Season == "Winter")), d = es$estimate)
  wil.fam <- rbind(wil.fam, data.frame(taxon = rownames(fam)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.fam$fdr <- mt(abx.fam, classlabel = "Season")$fdr



# Genus
abx.gen <- tax_glom(abx.use, taxrank="Genus")
gen <- data.frame(otu_table(abx.gen))
wil.gen <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(abx.gen)))[1]){
  es <- cohen.d(as.numeric(gen[taxon,]) ~ meta.abx$Season)
  pwr <- pwr.t2n.test(n1 = length(which(meta.abx$Season == "Summer")),
            n2 = length(which(meta.abx$Season == "Winter")), d = es$estimate)
  wil.gen <- rbind(wil.gen, data.frame(taxon = rownames(gen)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.gen$fdr <- mt(abx.gen, classlabel = "Season")$fdr



# OTU
otus <- data.frame(otu_table(abx.use))
wil.spe <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(abx.use)))[1]){
  es <- cohen.d(as.numeric(otus[taxon,]) ~ meta.abx$Season)
  pwr <- pwr.t2n.test(n1 = length(which(meta.abx$Season == "Summer")),
            n2 = length(which(meta.abx$Season == "Winter")), d = es$estimate)
  wil.spe <- rbind(wil.spe, data.frame(taxon = rownames(otus)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.spe$fdr <- mt(abx.use, classlabel = "Season")$fdr


```


### ABX Microbes - Create tables
```{r}
#barplot(sort(taxa_sums(phy.sum.c.abx.rem),TRUE)[1:10]/nsamples(phy.sum.c.abx.rem),las=2,cex.axis=.7)

phy.sum.c.abx.df <- psmelt(phy.sum.c.abx.rem)
phy.sum.c.abx.sum <- phy.sum.c.abx.df %>%
  group_by(OTU, Genus) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))


phy.sum.m.abx.df <- psmelt(phy.sum.m.abx.rem)
phy.sum.m.abx.sum <- phy.sum.m.abx.df %>%
  group_by(OTU, Genus) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))



phy.win.c.abx.df <- psmelt(phy.win.c.abx.rem)
phy.win.c.abx.sum <- phy.win.c.abx.df %>%
  group_by(OTU, Genus) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))




#write.csv(phy.sum.c.abx.sum, "16S/SummerContentABX_TaxTable.csv")
#write.csv(phy.sum.m.abx.sum, "16S/SummerMucosaABX_TaxTable.csv")
#write.csv(phy.win.c.abx.sum, "16S/WinterContentABX_TaxTable.csv")
```


### ABX Microbes - All samples
```{r}
phy.c.abx <- subset_samples(phy.c, Antibiotics == "Yes")
phy.c.abx <- filter_taxa(phy.c.abx, function(x) sum(x) > 0, T)


phy.c.abx.df <- psmelt(phy.c.abx)
phy.c.abx.sum <- phy.c.abx.df %>%
  group_by(OTU, Genus, Season) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))

#write.csv(phy.c.abx.sum, "16S/ContentABX_TaxTable.csv")


### Effect size calculation
cohen.d(phy.c.abx.df$Abundance ~ phy.c.abx.df$Season)
  # d = -8.822391e-18 (negligible)


### Power calculation
pwr.t2n.test(n1 = length(which(phy.c.abx.df$Season == "Summer")),
            n2 = length(which(phy.c.abx.df$Season == "Winter")), d = 8.591606e-18)
  # power = 0.05


### Species test
otus <- data.frame(otu_table(phy.c.abx))
wil.spe <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(phy.c.abx)))[1]){
  wil <- wilcox.test(x=as.numeric(otus[taxon,]), g=meta$Season)
  wil.spe <- rbind(wil.spe, data.frame(taxon = rownames(otus)[taxon], pval = wil$p.value))
}
wil.spe$padj <- p.adjust(wil.spe$pval, method = "BH")




### Identify top 10 taxa in summer and winter content ABX
phy.c.abx.sum.top <- phy.c.abx.sum[which(phy.c.abx.sum$Season == "Summer"),]
phy.c.abx.sum.top <- phy.c.abx.sum.top[order(-phy.c.abx.sum.top$Mean),]
s.top <- phy.c.abx.sum.top[1:10,]

phy.c.abx.win.top <- phy.c.abx.sum[which(phy.c.abx.sum$Season == "Winter"),]
phy.c.abx.win.top <- phy.c.abx.win.top[order(-phy.c.abx.win.top$Mean),]
w.top <- phy.c.abx.win.top[1:10,]





## Subset out top 10 winter taxa from all samples
phy.c.abx.top <- subset_taxa(phy.c.abx, Species == w.top$OTU[1] | Species == w.top$OTU[2] |
                               Species == w.top$OTU[3] | Species == w.top$OTU[4] |
                               Species == w.top$OTU[5] | Species == w.top$OTU[6] |
                               Species == w.top$OTU[7] | Species == w.top$OTU[8] |
                               Species == w.top$OTU[9] | Species == w.top$OTU[10])
phy.c.abx.top.df <- psmelt(phy.c.abx.top)
```


### ABX Microbes - Plot top ABX taxa
```{r}
ggplot(phy.c.abx.top.df, aes(x=Season, y=Abundance, fill=Season)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize = 1.5) +
  facet_wrap(OTU ~ ., ncol = 5, scales = "free") +
  scale_fill_manual(name="Season", values=c("#E69F00", "#0072B2")) +
  ylab("Relative Abundance") +
  stat_summary(fun.data = "mean_se", fun.args = list(mult = 1), geom = "pointrange",
               color = "black", position = position_dodge(0.7), size = 0.75,
               show.legend = F) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.text.x = element_text(size=14),
        axis.text.y = element_text(size=14),
        axis.title.x = element_text(size=20, margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y = element_text(size=20, margin=margin(t=0, r=10, b=0, l=0)),
        strip.text.x = element_text(size=20),
        strip.text.y = element_text(size=14),
        panel.spacing=unit(1, "lines"),
        legend.title = element_text(size = 18, face="bold"),
        legend.text = element_text(size=16),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
  ) 
```
### Alpha Div
```{r}
adiv.c <- data.frame(sample_data(phy.c))
adiv.c.noabx <- adiv.c[which(adiv.c$Antibiotics == "No"),]
adiv.c.noabx.act <- adiv.c.noabx[which(adiv.c.noabx$Season != "Winter"),]
  # Removing lone Winter sample left


### Chao ###
hist(adiv.c.noabx.act$chao, breaks=10)
qqnorm(adiv.c.noabx.act$chao)
qqline(adiv.c.noabx.act$chao)
shapiro.test(adiv.c.noabx.act$chao)
  # p-value = 0.8795
  # Normal

t.test(chao ~ Season, adiv.c.noabx.act)
  # p-value = 0.2791

var.test(chao ~ Season, adiv.c.noabx.act)
  # p-value = 0.6088



### Shannon ###
hist(adiv.c.noabx.act$shannon, breaks=10)
qqnorm(adiv.c.noabx.act$shannon)
qqline(adiv.c.noabx.act$shannon)
shapiro.test(adiv.c.noabx.act$shannon)
  # p-value = 0.008157
  # Not Normal

wilcox.test(shannon ~ Season, adiv.c.noabx.act)
  # p-value = 0.3817

fligner.test(shannon ~ Season, adiv.c.noabx.act)
  # p-value = 0.4269



### Inverse Simpson ###
hist(adiv.c.noabx.act$invsimpson, breaks=10)
qqnorm(adiv.c.noabx.act$invsimpson)
qqline(adiv.c.noabx.act$invsimpson)
shapiro.test(adiv.c.noabx.act$invsimpson)
  # p-value = 0.07148
  # Normal

t.test(invsimpson ~ Season, adiv.c.noabx.act)
  # p-value = 0.1551

var.test(invsimpson ~ Season, adiv.c.noabx.act)
  # p-value = 0.2264
```
### Beta Div
```{r}
### Bray-Curtis ###
phy.act.noabx <- subset_samples(phy.c, Antibiotics == "No")
phy.act.noabx <- subset_samples(phy.act.noabx, Season != "Winter")

bray.pcoa <- ordinate(physeq=phy.act.noabx, method="PCoA", distance="bray")

# Test
sampdf <- data.frame(sample_data(phy.act.noabx))
bray <- phyloseq::distance(physeq = phy.act.noabx, method = "bray")
beta.tax <- betadisper(d = bray, group = sampdf$Season)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.803
adonis(bray ~ Season, data=sampdf, permutations = 9999)
  # p-value = 0.1385

```

### Test Diet
```{r}
phy.diet <- subset_samples(phy.c, Individual == "S03" | Individual == "S05" | Individual == "S06" | Individual == "S07" | Individual == "S08" | Individual == "S10" | Individual == "S11" | Individual == "S12" | Individual == "S21" | Individual == "S23" | Individual == "S24" | Individual == "S29"  | Individual == "S30" | Individual == "S32" | Individual == "S34" | Individual == "S39" | Individual == "S47" | Individual == "S48" | Individual == "S49" | Individual == "S50")


adiv.diet <- data.frame(sample_data(phy.diet))


### Chao ###
hist(adiv.diet$chao, breaks=10)
qqnorm(adiv.diet$chao)
qqline(adiv.diet$chao)
shapiro.test(adiv.diet$chao)
  # p-value = 0.3446
  # Normal

t.test(chao ~ Diet, adiv.diet)
  # p-value = 0.08385

var.test(chao ~ Diet, adiv.diet)
  # p-value = 0.6259



### Shannon ###
hist(adiv.diet$shannon, breaks=10)
qqnorm(adiv.diet$shannon)
qqline(adiv.diet$shannon)
shapiro.test(adiv.diet$shannon)
  # p-value = 0.01137
  # Not Normal

wilcox.test(shannon ~ Diet, adiv.diet)
  # p-value = 0.02398*

fligner.test(shannon ~ Diet, adiv.diet)
  # p-value = 0.08924



### Inverse Simpson ###
hist(adiv.diet$invsimpson, breaks=10)
qqnorm(adiv.diet$invsimpson)
qqline(adiv.diet$invsimpson)
shapiro.test(adiv.diet$invsimpson)
  # p-value = 0.2037
  # Normal

t.test(invsimpson ~ Diet, adiv.diet)
  # p-value = 0.06893

var.test(invsimpson ~ Diet, adiv.diet)
  # p-value = 0.8414





### Bray-Curtis ###
bray.pcoa <- ordinate(physeq=phy.diet, method="PCoA", distance="bray")

# Test
sampdf <- data.frame(sample_data(phy.diet))
bray <- phyloseq::distance(physeq = phy.diet, method = "bray")
beta.tax <- betadisper(d = bray, group = sampdf$Diet)
p <- permutest(beta.tax)
p$tab
  # p-value = 0.024
adonis(bray ~ Diet, data=sampdf, permutations = 9999)
  # p-value = 0.1621
```

### Diet - RTOST
```{r}
# Code modified from Skarlupka, et al. (2019)
# https://doi.org/10.1186/s40104-019-0375-0


# Format OTU table
TOST.otu <- data.frame(t(data.frame(otu_table(phy.diet))))
TOST.otu$Sample <- rownames(TOST.otu)
TOST.otu$Diet <- "Diet 1"
diet2 <- c("CC1_S47", "CC1_S48", "CC1_S49", "CC1_S50")
for (diet_samp in 1:length(diet2)){
  x <- which(TOST.otu$Sample == diet2[diet_samp])
  TOST.otu$Diet[x] <- "Diet 2"
}


# Separate by Diet
TOST.otu.diet1 <- TOST.otu[which(TOST.otu$Diet == "Diet 1"),]
TOST.otu.diet2 <- TOST.otu[which(TOST.otu$Diet == "Diet 2"),]


# Calculate mean OTUs for diet 1 and pull out diet 2 samples; remove Sample and Diet cols I made
TOST.otu.diet1.mean <- aggregate(TOST.otu.diet1[ , 1: ( ncol(TOST.otu.diet1)-2 ) ],
                               list(TOST.otu.diet1$Diet), mean)
TOST.otu.diet1.mean <- TOST.otu.diet1.mean[-1]
TOST.S47 <- TOST.otu.diet2[1,]
TOST.S48 <- TOST.otu.diet2[2,]
TOST.S49 <- TOST.otu.diet2[3,]
TOST.S50 <- TOST.otu.diet2[4,]


# Prepare data
TOST.otu.diet1.t <- t(as.matrix(sapply(TOST.otu.diet1.mean[,1:ncol(TOST.otu.diet1.mean)],
                                       as.numeric)))
TOST.S47.t <- t(as.matrix(sapply(TOST.S47[,1:ncol(TOST.otu.diet1.mean)], as.numeric)))
TOST.S48.t <- t(as.matrix(sapply(TOST.S48[,1:ncol(TOST.otu.diet1.mean)], as.numeric)))
TOST.S49.t <- t(as.matrix(sapply(TOST.S49[,1:ncol(TOST.otu.diet1.mean)], as.numeric)))
TOST.S50.t <- t(as.matrix(sapply(TOST.S50[,1:ncol(TOST.otu.diet1.mean)], as.numeric)))


### Explore Data
summary(as.vector(TOST.otu.diet1.t))
length(which(TOST.otu.diet1.t > 100, arr.ind = T))
  # 0 values above 100
jj <- scale(as.vector(TOST.otu.diet1.t))
qqnorm(jj, main = "QQ-Plot for Summer No ABX Diet 1")
abline(0, 1, col="red")
  # Not Normal
plot(density(as.vector(TOST.otu.diet1.t)), main="Density Plot for Summer No ABX Diet 1")
  # Skewed right
shapiro.test(TOST.otu.diet1.t)
  # p-value  < 2.2e-16

summary(as.vector(TOST.S47.t))
length(which(TOST.S47.t > 100, arr.ind = T))
  # 0 values above 100
jj <- scale(as.vector(TOST.S47.t))
qqnorm(jj, main = "QQ-Plot for S47 Cecal Content")
abline(0, 1, col="red")
  # Not Normal
plot(density(as.vector(TOST.S47.t)), main="Density Plot for S47 Cecal Content")
  # Skewed right
shapiro.test(TOST.S47.t)
  # p-value  < 2.2e-16

summary(as.vector(TOST.S48.t))
length(which(TOST.S48.t > 100, arr.ind = T))
  # 0 values above 100
jj <- scale(as.vector(TOST.S48.t))
qqnorm(jj, main = "QQ-Plot for S48 Cecal Content")
abline(0, 1, col="red")
  # Not Normal
plot(density(as.vector(TOST.S48.t)), main="Density Plot for S48 Cecal Content")
  # Skewed right
shapiro.test(TOST.S48.t)
  # p-value  < 2.2e-16

summary(as.vector(TOST.S49.t))
length(which(TOST.S49.t > 100, arr.ind = T))
  # 0 values above 100
jj <- scale(as.vector(TOST.S49.t))
qqnorm(jj, main = "QQ-Plot for S49 Cecal Content")
abline(0, 1, col="red")
  # Not Normal
plot(density(as.vector(TOST.S49.t)), main="Density Plot for S49 Cecal Content")
  # Skewed right
shapiro.test(TOST.S49.t)
  # p-value  < 2.2e-16

summary(as.vector(TOST.S50.t))
length(which(TOST.S50.t > 100, arr.ind = T))
  # 0 values above 100
jj <- scale(as.vector(TOST.S50.t))
qqnorm(jj, main = "QQ-Plot for S50 Cecal Content")
abline(0, 1, col="red")
  # Not Normal
plot(density(as.vector(TOST.S50.t)), main="Density Plot for S50 Cecal Content")
  # Skewed right
shapiro.test(TOST.S50.t)
  # p-value  < 2.2e-16


### Because data isn't Normally distributed, I'll use RTOST instead of TOST ###
### RTOST ###
rtost(TOST.otu.diet1.t, TOST.S47.t, alpha = 0.05)
  # p-value = 0
  # Reject hypothesis that random & mean have non-equivalent means
rtost(TOST.otu.diet1.t, TOST.S48.t, alpha = 0.05)
  # p-value = 0
  # Reject hypothesis that random & mean have non-equivalent means
rtost(TOST.otu.diet1.t, TOST.S49.t, alpha = 0.05)
  # p-value = 0
  # Reject hypothesis that random & mean have non-equivalent means
rtost(TOST.otu.diet1.t, TOST.S50.t, alpha = 0.05)
  # p-value = 0
  # Reject hypothesis that random & mean have non-equivalent means
```