---
title: "Code_AllShortReads"
author: "Edna Chiang"
date: "2/8/2022"
output: html_document
---

### Load Packages, scripts, and functions
```{r}
library(ggplot2)
library(plyr)
library(dplyr)
library(grid)
library(phyloseq)
library(vegan)
library(car)
library(reshape2)
source('scripts/misc.R')
source('scripts/simper_pretty.R')
source('scripts/R_krusk.R')
source('scripts/removeTaxa.R')
source('scripts/kaijuCreatePhyseq.R')
set.seed(1)
theme_set(theme_bw())
```



### DON'T RUN - Prepare taxonomic data
```{r}
### Remove taxa in < 3 samples
#removeTaxa('kaiju/all_shortreads/')

### Create phyloseq objects
#kaijuCreatePhyseq('kaiju/all_shortreads/', 'metadata_metag.csv')
```

### Load tax physeq data
```{r}
### Load physeq data
load("kaiju/all_shortreads/use/physeq/phylum.RData")
phy.phy <- physeq
load("kaiju/all_shortreads/use/physeq/class.RData")
cla.phy <- physeq
load("kaiju/all_shortreads/use/physeq/order.RData")
ord.phy <- physeq
load("kaiju/all_shortreads/use/physeq/family.RData")
fam.phy <- physeq
load("kaiju/all_shortreads/use/physeq/genus.RData")
gen.phy <- physeq
load("kaiju/all_shortreads/use/physeq/species.RData")
spe.phy <- physeq

### Remove W20
phy.phy <- subset_samples(phy.phy, ID != "S.W20")
cla.phy <- subset_samples(cla.phy, ID != "S.W20")
ord.phy <- subset_samples(ord.phy, ID != "S.W20")
fam.phy <- subset_samples(fam.phy, ID != "S.W20")
gen.phy <- subset_samples(gen.phy, ID != "S.W20")
spe.phy <- subset_samples(spe.phy, ID != "S.W20")



### Melt physeq objects into dataframes
phy.melt <- psmelt(phy.phy)
cla.melt <- psmelt(cla.phy)
ord.melt <- psmelt(ord.phy)
fam.melt <- psmelt(fam.phy)
gen.melt <- psmelt(gen.phy)
spe.melt <- psmelt(spe.phy)


### Import metadata
meta <- read.csv('metadata_metag.csv', header=T)
meta$Season <- ordered(meta$Season, levels=c("Summer", "Winter", "Spring"))
```


### Summarize tax data
```{r}
phy.sum <- phy.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(phy.sum, file = "kaiju/all_shortreads/phylum.sum.csv", row.names = F)

cla.sum <- cla.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(cla.sum, file = "kaiju/all_shortreads/class.sum.csv", row.names = F)

ord.sum <- ord.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(ord.sum, file = "kaiju/all_shortreads/order.sum.csv", row.names = F)

fam.sum <- fam.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(fam.sum, file = "kaiju/all_shortreads/family.sum.csv", row.names = F)

gen.sum <- gen.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(gen.sum, file = "kaiju/all_shortreads/genus.sum.csv", row.names = F)

spe.sum <- spe.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(spe.sum, file = "kaiju/all_shortreads/species.sum.csv", row.names = F)
```


### Taxa Broad Stats - Paper
```{r}
### Load Data
domain <- read.csv("kaiju/all_shortreads/use/tables/domain.csv", row.names = 1)
dom.otu <- otu_table(domain, taxa_are_rows = T)
dom.tax <- data.frame(rownames(domain))
dom.tax <- tax_table(dom.tax)
taxa_names(dom.tax) <- taxa_names(dom.otu)
dom.meta <- sample_data(meta)
sample_names(dom.meta) <- meta$ID
dom.phy <- merge_phyloseq(dom.otu, dom.tax, dom.meta)

# Remove outlier
dom.phy <- subset_samples(dom.phy, ID != "S.W20")

dom.melt <- psmelt(dom.phy)

dom.sum <- dom.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))

dom.overall <- dom.melt %>%
  group_by(OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance),
            Min = min(Abundance),
            Max = max(Abundance))

phy.overall <- phy.melt %>%
  group_by(OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance),
            Min = min(Abundance),
            Max = max(Abundance))
#write.csv(phy.overall, file = "phy.overall.csv", row.names = F)

#write.csv(dom.sum, file = "kaiju/all_shortreads/use/summarized/domain.sum.csv", row.names = F)
```

### Sample dist PCoA & PERMANOVA
```{r}
bray.pcoa <- ordinate(physeq=spe.phy, method="PCoA", distance="bray")
sample_data(spe.phy)$Season <- ordered(sample_data(spe.phy)$Season, levels = c("Summer", "Winter", "Spring"))

#tiff("aaas.pcoa.tiff", width=8, height=6, units="in", res=300)
plot_ordination(physeq = spe.phy, ordination = bray.pcoa, axes = c(1,2),
                color = "Season") +
  geom_point(size=10) +
  scale_color_manual(values=c("#00cccc", "#008484", "#004f4f")) +
  ggtitle("CAZyme Profiles") +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.title.x = element_text(size=20),
        axis.title.y = element_text(size=20),
        legend.title = element_text(size=20),
        legend.text = element_text(size=18),
        plot.title = element_blank())
#dev.off()

```



### DNA Yield
```{r}
dna <- read.csv('dna.csv')
dna.c <- dna[which(grepl("CC", dna$Sample) == T),]
dna.c$Metabolism <- "a"
dna.c$Metabolism[which(dna.c$Season == "Winter")] <- "Hibernation"
dna.c$Metabolism[which(dna.c$Season != "Winter")] <- "Active"
dna.c$Metabolism <- ordered(dna.c$Metabolism, levels = c("Active", "Hibernation"))
dna.c$Season <- ordered(dna.c$Season, levels = c("Summer", "Winter", "Spring"))

# Remove outliers
dna.c <- dna.c[-which(dna.c$Squirrel == "S23"),]
dna.c <- dna.c[-which(dna.c$Squirrel == "S37"),]
dna.c <- dna.c[-which(dna.c$Squirrel == "S60"),]
dna.c <- dna.c[-which(dna.c$Squirrel == "W43"),]
dna.c <- dna.c[-which(dna.c$Squirrel == "W20"),]
  # This is due to odd metagenome
dna.c <- dna.c[-which(dna.c$Squirrel == "W50"),]
dna.c <- dna.c[-which(dna.c$Squirrel == "P20"),]
dna.c <- dna.c[-which(dna.c$Squirrel == "P22"),]

dna.c.noabx <- dna.c[which(dna.c$Antibiotics == "No Antibiotics"),]


shapiro.test(dna.c$DNA_Yield_.ng.mg.)
  # p-value = 4.583e-08
  # Not Normal
shapiro.test(dna.c.noabx$DNA_Yield_.ng.mg.)
  # p-value = 0.601
  # Normal

summary(aov(formula = DNA_Yield_.ng.mg. ~ Season, data = dna.c))
  # p-value = 0.411
summary(aov(formula = DNA_Yield_.ng.mg. ~ Season, data = dna.c.noabx))
  # p-value = 0.0472*
TukeyHSD(aov(formula = DNA_Yield_.ng.mg. ~ Season, data = dna.c.noabx))
  # Summer-Winter = 0.9023267
  # Summer-Spring = 0.0499387*
  # Winter-Spring = 0.1449934

t.test(formula = DNA_Yield_.ng.mg. ~ Metabolism, data = dna.c.noabx)
  # p-value = 0.3962
```



### Plot - DNA Yield
```{r}
#dna.c$SeasonABX <- interaction(dna.c$Season, dna.c$Antibiotics)
#dna.c$SeasonABX <- ordered(dna.c$SeasonABX, levels=c("Summer.No Antibiotics", "Summer.Antibiotics", "Winter.No Antibiotics", "Winter.Antibiotics", "Spring.No Antibiotics", "Spring.Antibiotics"))



tiff("dna_yield.tiff", width = 5, height = 2.5, units = "in", res = 600)
ggplot(dna.c, aes(x=Season, y=DNA_Yield_.ng.mg., colour = SeasonABX, fill = SeasonABX)) +
  facet_grid( . ~ Substrate) +
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize = 1.5,
               position = position_dodge(0.7), stackratio = 0.5) +
  scale_color_manual(values = c("#c98b00", "#E69F00", "#004166", "#006299", "#00966d",
                                "#00bf8b")) +
  scale_fill_manual(values = c("#E69F00", "white", "#006299", "white", "#00bf8b",
                               "white")) +
  ylab("DNA Yield\n(ng/mg cecal content)") +
  scale_y_continuous(breaks=c(0.0, 51.5, 103, 154.5, 206),
                     limits=c(-3, 206)) +
  stat_summary(fun.data = "mean_cl_normal", geom = "pointrange",
               color = "black", position = position_dodge(0.7), size = 0.4,
               show.legend = F) +
  theme(plot.title=element_text(hjust=0.5, size=12),
        axis.text.x = element_text(size=10),
        axis.text.y = element_text(size=10),
        axis.title.x = element_text(size=10, margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y = element_text(size=10, margin=margin(t=0, r=10, b=0, l=0)),
        strip.text.x = element_text(size=11, face = "bold"),
        strip.text.y = element_text(size=10),
        strip.background = element_blank(),
        panel.spacing=unit(1, "lines"),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        legend.position = "None"
  ) 
dev.off()
```






### Test host DNA yield
```{r}
### Calculate Host DNA Yield
microbe.ra$Host_DNA_Yield <- microbe.ra$Host_Percentage * microbe.ra$DNAyield



### Test Host DNA Yield
shapiro.test(microbe.ra$Host_DNA_Yield)
  # p-value = 5.961e-08
  # Not Normal

kruskal.test(formula = Host_DNA_Yield ~ Season, data = microbe.ra)
  # p-value = 0.1129
wilcox.test(formula = Host_DNA_Yield ~ Metabolism, data = microbe.ra)



### Test Host DNA Yield Variance
fligner.test(Host_DNA_Yield ~ Season, microbe.ra)
  # p-value = 0.002298*
fligner.test(Host_DNA_Yield ~ Season, microbe.ra, subset = (Season != "Spring") )
  # Summer vs. Winter
  # p-value = 0.005881*
fligner.test(Host_DNA_Yield ~ Season, microbe.ra, subset = (Season != "Winter") )
  # Summer vs. Spring
  # p-value = 0.8869
fligner.test(Host_DNA_Yield ~ Season, microbe.ra, subset = (Season != "Summer") )
  # Winter vs. Spring
  # p-value = 0.005881*
```