---
title: "Code_Inulinase"
author: "Edna Chiang"
date: "12/14/2021"
output: html_document
---

### Load Packages, scripts, and functions
```{r}
library(ggplot2)
library(plyr)
library(dplyr)
library(grid)
#library(MESS)
#library(pgirmess)
#library(Bolstad2)
library(phyloseq)
library(vegan)
#library(miLineage)
#library(psych)
library(car)
library(reshape2)
library(pwr)
library(effectsize)
#source('scripts/import_crds.R')
#source('scripts/bin_crds.R')
source('scripts/misc.R')
#source('scripts/calc_delta_metrics.R')
source('scripts/simper_pretty.R')
source('scripts/R_krusk.R')
source('scripts/removeTaxa.R')
set.seed(1)
theme_set(theme_bw())
```



### Average genome size /  Genome Equivalents
```{r}
ags <- read.csv('AverageGenomeSize.csv')

# W20 is an outlier (very large AGS-- suggests more generalists)
ags <- ags[-which(ags$Sample == "S-W20"),]
ags$Season <- ordered(ags$Season, levels=c("Summer", "Winter", "Spring"))

# Convert AGS to kb
ags$AGS_kb <- ags$Avg_Genome_Size / 1000

shapiro.test(ags$Avg_Genome_Size)
  # p-value = 0.3697
  # Normal
shapiro.test(ags$Genome_Equivalents)
  # p-value = 0.2279
  # Normal


summary(aov(Avg_Genome_Size ~ Season, data = ags))
  # p-value = 0.00868
TukeyHSD(aov(Avg_Genome_Size ~ Season, data = ags))
  # Summer-Winter = 0.0104046
  # Summer-Spring = 0.8820160
  # Winter-Spring = 0.0250999
summary(aov(Genome_Equivalents ~ Season, data = ags))
  # p-value = 0.325

#t.test(Avg_Genome_Size ~ Metabolism, data = ags)
  # p-value = 0.008007
#t.test(Genome_Equivalents ~ Metabolism, data = ags)
  # p-value = 0.4634



```

### Plot - AGS
```{r}
#tiff("inulinase.ags.tiff", width = 175, height = 105, units = "mm", res = 400)
ggplot(ags, aes(x = Season, y = AGS_kb, fill = Season)) +
  geom_dotplot(binaxis = 'y', stackdir = 'center', dotsize = 1.5, stackratio = 1.15) +
  scale_fill_manual(values = c("#E69F00", "#0072B2", "#00bf8b")) +
  scale_y_continuous(limits = c(2690, 4000), breaks = c(2690, 2952, 3214, 3476, 3738, 4000)) +
  stat_summary(fun.data = "mean_se", fun.args = list(mult = 1), geom = "pointrange",
               position = position_dodge(0.7), size = 0.75,
               show.legend = F) +
  ylab("Average Genome Size (kb)") +
    theme(plot.title=element_text(hjust=0.5, size=20),
        axis.text.x = element_text(size=12),
        axis.text.y = element_text(size=12),
        axis.title.x = element_text(size=20, margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y = element_text(size=20, margin=margin(t=0, r=10, b=0, l=0)),
        panel.spacing=unit(1, "lines"),
        legend.title = element_text(size = 18, face="bold"),
        legend.text = element_text(size=16),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
  ) 
#dev.off()
```

### Endo/Exo Rel Abund & RPKG
```{r}
### Rel Abund
mapped_all <- read.csv('eCAMI/mapped_reads_all.csv')
exo_mapped <- mapped_all[which(mapped_all$Protein == "Exo"),]
endo_mapped <- mapped_all[which(mapped_all$Protein == "Endo"),]

# Remove outlier W20 (not in endo)
exoMapped <- exo_mapped[-which(exo_mapped$Sample == "S-W20"),]

# Normality
shapiro.test(exoMapped$RelAbund.100000)
  # p-value = 0.4807
  # Normal
shapiro.test(endo_mapped$RelAbund.100000)
  # p-value = 3.281e-06
  # Not Normal

# Kruskal-Wallis
summary(aov(RelAbund.100000 ~ Season, data = exoMapped))
  # p-value = 0.992
t.test(RelAbund.100000 ~ Metabolism, data = exoMapped)
  # p-value = 0.979
  
  
kruskal.test(formula = RelAbund.100000 ~ Season, data = endo_mapped)
  # p-value = 0.3329
wilcox.test(formula = RelAbund.100000 ~ Metabolism, data = endo_mapped)
  # p-value = 0.5

# Variance
leveneTest(RelAbund.100000 ~ Season, data=exoMapped)
  # p-value = 0.3954
leveneTest(RelAbund.100000 ~ Season, data=endo_mapped)
  # p-value = 0.4414
leveneTest(RelAbund.100000 ~ Metabolism, data=exoMapped)
  # p-value = 0.8994
leveneTest(RelAbund.100000 ~ Metabolism, data=endo_mapped)
  # p-value = 0.5622



### Exo - RPKG
exo_rpkg <- read.csv('eCAMI/exoRPKG.csv')
exo_rpkg$Season <- ordered(exo_rpkg$Season, c("Summer", "Winter", "Spring"))

# Remove W20 outlier
exo_rpkg <- exo_rpkg[-which(exo_rpkg$Sample == "S-W20"),]


shapiro.test(exo_rpkg$RPKG)
  # p-value = 0.8887

summary(aov(formula = RPKG ~ Season, data = exo_rpkg))
  # p-value = 0.598
t.test(formula = RPKG ~ Metabolism, data = exo_rpkg)
  # p-value = 0.3374

fligner.test(RPKG ~ Season, data=exo_rpkg)
  # p-value = 0.1426
fligner.test(RPKG ~ Metabolism, data=exo_rpkg)
  # p-value = 0.7648
```


### Plot - RPKG
```{r}
#tiff("inulinase.rpkg.tiff", width = 175, height = 100, units = "mm", res = 400)
ggplot(exo_rpkg, aes(x=Season, y=RPKG, fill=Season)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize = 1.5) +
  scale_fill_manual(name="Season", values=c("#E69F00", "#0072B2", "#00bf8b")) +
  ylab("Reads per kilobase\nper genome equivalent") +
  scale_y_continuous(breaks=c(0.00, 0.142, 0.284, 0.426, 0.568, 0.710),
                     limits=c(0.000, 0.710)) +
  stat_summary(fun.data = "mean_se", fun.args = list(mult = 1), geom = "pointrange",
               color = "black", position = position_dodge(0.7), size = 0.75,
               show.legend = F) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.text.x = element_text(size=14),
        axis.text.y = element_text(size=14),
        axis.title.x = element_text(size=20, margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y = element_text(size=20, margin=margin(t=0, r=10, b=0, l=0)),
        strip.text.x = element_text(size=20),
        strip.text.y = element_text(size=14),
        panel.spacing=unit(1, "lines"),
        legend.title = element_text(size = 18, face="bold"),
        legend.text = element_text(size=16),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()
  ) 
#dev.off()




```



### DON'T RUN - Prepare taxonomic data
```{r}
### Remove taxa found in < 3 samples
#removeTaxa('kaiju/inulinase/relAbund_tables/')


### Import rel abund tables
phy.ra <- read.csv('kaiju/inulinase/relAbund_tables/exo.phylum.csv', header=T, row.names=1)
cla.ra <- read.csv('kaiju/inulinase/relAbund_tables/exo.class.csv', header=T, row.names=1)
ord.ra <- read.csv('kaiju/inulinase/relAbund_tables/exo.order.csv', header=T, row.names=1)
fam.ra <- read.csv('kaiju/inulinase/relAbund_tables/exo.family.csv', header=T, row.names=1)
gen.ra <- read.csv('kaiju/inulinase/relAbund_tables/exo.genus.csv', header=T, row.names=1)
spe.ra <- read.csv('kaiju/inulinase/relAbund_tables/exo.species.csv', header=T, row.names=1)

phy.otu <- otu_table(phy.ra, taxa_are_rows = T)
cla.otu <- otu_table(cla.ra, taxa_are_rows = T)
ord.otu <- otu_table(ord.ra, taxa_are_rows = T)
fam.otu <- otu_table(fam.ra, taxa_are_rows = T)
gen.otu <- otu_table(gen.ra, taxa_are_rows = T)
spe.otu <- otu_table(spe.ra, taxa_are_rows = T)


### Import metadatac
meta <- read.csv('metadata_metag.csv', header=T)
meta <- meta[-which(meta$ID == "S.W20"),]
meta.samp <- sample_data(meta)
sample_names(meta.samp) <- meta$ID
meta.samp$Season <- ordered(meta$Season, levels=c("Summer", "Winter", "Spring"))


### Create tax table
phy <- data.frame(full = taxa_names(phy.otu))
phy <- cbind(phy,colsplit(phy$full, pattern = ";", names = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")))
rownames(phy) <- phy$full
phy <- phy[,-1]

cla <- data.frame(full = taxa_names(cla.otu))
cla <- cbind(cla,colsplit(cla$full, pattern = ";", names = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")))
rownames(cla) <- cla$full
cla <- cla[,-1]

ord <- data.frame(full = taxa_names(ord.otu))
ord <- cbind(ord,colsplit(ord$full, pattern = ";", names = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")))
rownames(ord) <- ord$full
ord <- ord[,-1]

fam <- data.frame(full = taxa_names(fam.otu))
fam <- cbind(fam,colsplit(fam$full, pattern = ";", names = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")))
rownames(fam) <- fam$full
fam <- fam[,-1]

gen <- data.frame(full = taxa_names(gen.otu))
gen <- cbind(gen,colsplit(gen$full, pattern = ";", names = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")))
rownames(gen) <- gen$full
gen <- gen[,-1]

spe <- data.frame(full = taxa_names(spe.otu))
spe <- cbind(spe,colsplit(spe$full, pattern = ";", names = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species")))
rownames(spe) <- spe$full
spe <- spe[,-1]

phy.tax <- tax_table(phy)
cla.tax <- tax_table(cla)
ord.tax <- tax_table(ord)
fam.tax <- tax_table(fam)
gen.tax <- tax_table(gen)
spe.tax <- tax_table(spe)


### Make phyloseq objects
taxa_names(phy.tax) <- taxa_names(phy.otu)
phy.phy <- merge_phyloseq(phy.tax, phy.otu, meta.samp)
taxa_names(cla.tax) <- taxa_names(cla.otu)
cla.phy <- merge_phyloseq(cla.tax, cla.otu, meta.samp)
taxa_names(ord.tax) <- taxa_names(ord.otu)
ord.phy <- merge_phyloseq(ord.tax, ord.otu, meta.samp)
taxa_names(fam.tax) <- taxa_names(fam.otu)
fam.phy <- merge_phyloseq(fam.tax, fam.otu, meta.samp)
taxa_names(gen.tax) <- taxa_names(gen.otu)
gen.phy <- merge_phyloseq(gen.tax, gen.otu, meta.samp)
taxa_names(spe.tax) <- taxa_names(spe.otu)
spe.phy <- merge_phyloseq(spe.tax, spe.otu, meta.samp)


### Save physeq objects
#save(phy.phy, file="kaiju/inulinase/physeq/phylum.phy.RData")
#save(cla.phy, file="kaiju/inulinase/physeq/class.phy.RData")
#save(ord.phy, file="kaiju/inulinase/physeq/order.phy.RData")
#save(fam.phy, file="kaiju/inulinase/physeq/family.phy.RData")
#save(gen.phy, file="kaiju/inulinase/physeq/genus.phy.RData")
#save(spe.phy, file="kaiju/inulinase/physeq/species.phy.RData")
```

### Load tax physeq data
```{r}
### Load physeq data
load("kaiju/inulinase/physeq/phylum.phy.RData")
load("kaiju/inulinase/physeq/class.phy.RData")
load("kaiju/inulinase/physeq/order.phy.RData")
load("kaiju/inulinase/physeq/family.phy.RData")
load("kaiju/inulinase/physeq/genus.phy.RData")
load("kaiju/inulinase/physeq/species.phy.RData")


### Melt physeq objects into dataframes
phy.melt <- psmelt(phy.phy)
cla.melt <- psmelt(cla.phy)
ord.melt <- psmelt(ord.phy)
fam.melt <- psmelt(fam.phy)
gen.melt <- psmelt(gen.phy)
spe.melt <- psmelt(spe.phy)


### Import metadata
meta <- read.csv('metadata_metag.csv', header=T)
meta$Season <- ordered(meta$Season, levels=c("Summer", "Winter", "Spring"))
meta <- meta[-which(meta$ID == "S.W20"),]
```


### Summarize tax data - Season
```{r}
phy.sum <- phy.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(phy.sum, file = "kaiju/inulinase/tax_summarized/season/phylum.sum.csv", row.names = F)

cla.sum <- cla.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(cla.sum, file = "kaiju/inulinase/tax_summarized/season/class.sum.csv", row.names = F)

ord.sum <- ord.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(ord.sum, file = "kaiju/inulinase/tax_summarized/season/order.sum.csv", row.names = F)

fam.sum <- fam.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(fam.sum, file = "kaiju/inulinase/tax_summarized/season/family.sum.csv", row.names = F)

gen.sum <- gen.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(gen.sum, file = "kaiju/inulinase/tax_summarized/season/genus.sum.csv", row.names = F)

spe.sum <- spe.melt %>%
  group_by(Season, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(spe.sum, file = "kaiju/inulinase/tax_summarized/season/species.sum.csv", row.names = F)
```



### Summarize tax data - Metabolism
```{r}
phy.sum <- phy.melt %>%
  group_by(Metabolism, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(phy.sum, file = "kaiju/inulinase/tax_summarized/metabolism/phylum.sum.csv", row.names = F)

cla.sum <- cla.melt %>%
  group_by(Metabolism, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(cla.sum, file = "kaiju/inulinase/tax_summarized/metabolism/class.sum.csv", row.names = F)

ord.sum <- ord.melt %>%
  group_by(Metabolism, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(ord.sum, file = "kaiju/inulinase/tax_summarized/metabolism/order.sum.csv", row.names = F)

fam.sum <- fam.melt %>%
  group_by(Metabolism, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(fam.sum, file = "kaiju/inulinase/tax_summarized/metabolism/family.sum.csv", row.names = F)

gen.sum <- gen.melt %>%
  group_by(Metabolism, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(gen.sum, file = "kaiju/inulinase/tax_summarized/metabolism/genus.sum.csv", row.names = F)

spe.sum <- spe.melt %>%
  group_by(Metabolism, OTU) %>%
  summarize(Mean = mean(Abundance),
            SD = sd(Abundance),
            SE = se(Abundance))
#write.csv(spe.sum, file = "kaiju/inulinase/tax_summarized/metabolism/species.sum.csv", row.names = F)
```



### Tax - # of Taxa
```{r}
### Family-level
fam.phy.w <- subset_samples(fam.phy, Season == "Winter")
length(which(taxa_sums(fam.phy.w) == 0))
  # Winter has 7 fewer families than Summer and Spring

### Genus-level
gen.phy.w <- subset_samples(gen.phy, Season == "Winter")
length(which(taxa_sums(gen.phy.w) == 0))
  # Winter has 16 fewer genera than Summer and Spring

### Species-level
spe.phy.w <- subset_samples(spe.phy, Season == "Winter")
length(which(taxa_sums(spe.phy.w) == 0))
  # Winter has 41 fewer species than the whole dataset
spe.phy.s <- subset_samples(spe.phy, Season == "Summer")
length(which(taxa_sums(spe.phy.s) == 0))
  # Summer has 4 fewer species than the whole dataset
spe.phy.p <- subset_samples(spe.phy, Season == "Spring")
length(which(taxa_sums(spe.phy.p) == 0))
  # Spring has 8 fewer species than the whole dataset
spe.phy.act <- subset_samples(spe.phy, Metabolism == "Active")
length(which(taxa_sums(spe.phy.act) == 0))
  # Active seasons have 2 fewer species than the whole dataset
```



### Test taxa - Season
```{r}
# Phyla
otus <- data.frame(otu_table(phy.phy))
kw.phy <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(phy.phy)))[1]){
  kw <- kruskal.test(x=as.numeric(otus[taxon,]), g=meta$Season)
  es <- effectsize(kw)
  kw.phy <- rbind(kw.phy, data.frame(taxon = rownames(otus)[taxon], pval = kw$p.value, effect_size = es[[1]]))
}
kw.phy$padj <- p.adjust(kw.phy$pval, method = "BH")

# Class
otus <- data.frame(otu_table(cla.phy))
kw.cla <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(cla.phy)))[1]){
  kw <- kruskal.test(x=as.numeric(otus[taxon,]), g=meta$Season)
  kw.cla <- rbind(kw.cla, data.frame(taxon = rownames(otus)[taxon], pval = kw$p.value))
}
kw.cla$padj <- p.adjust(kw.cla$pval, method = "BH")

# Order
otus <- data.frame(otu_table(ord.phy))
kw.ord <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(ord.phy)))[1]){
  kw <- kruskal.test(x=as.numeric(otus[taxon,]), g=meta$Season)
  kw.ord <- rbind(kw.ord, data.frame(taxon = rownames(otus)[taxon], pval = kw$p.value))
}
kw.ord$padj <- p.adjust(kw.ord$pval, method = "BH")

# Family
otus <- data.frame(otu_table(fam.phy))
kw.fam <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(fam.phy)))[1]){
  kw <- kruskal.test(x=as.numeric(otus[taxon,]), g=meta$Season)
  kw.fam <- rbind(kw.fam, data.frame(taxon = rownames(otus)[taxon], pval = kw$p.value))
}
kw.fam$padj <- p.adjust(kw.fam$pval, method = "BH")

# Genus
otus <- data.frame(otu_table(gen.phy))
kw.gen <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(gen.phy)))[1]){
  kw <- kruskal.test(x=as.numeric(otus[taxon,]), g=meta$Season)
  kw.gen <- rbind(kw.gen, data.frame(taxon = rownames(otus)[taxon], pval = kw$p.value))
}
kw.gen$padj <- p.adjust(kw.gen$pval, method = "BH")

# Species
otus <- data.frame(otu_table(spe.phy))
kw.spe <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(spe.phy)))[1]){
  kw <- kruskal.test(x=as.numeric(otus[taxon,]), g=meta$Season)
  kw.spe <- rbind(kw.spe, data.frame(taxon = rownames(otus)[taxon], pval = kw$p.value))
}
kw.spe$padj <- p.adjust(kw.spe$pval, method = "BH")
```


### Test taxa - Metabolism
```{r}
### Prep
MetabOrder <- c(rep("Active",6), rep("Hibernating", 3), rep("Active", 6), rep ("Hibernating", 2))
  # Order that samples appear in OTU table
pwr.t2n.test(n1 = length(which(meta$Metabolism == "Active")), n2 = length(which(meta$Metabolism == "Hibernating")), power = 0.8)
  # d = 1.595744
  # Detectable effect size if power = 0.8



# Phyla
otus <- data.frame(otu_table(phy.phy))
wil.phy <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(phy.phy)))[1]){
  es <- cohen.d(as.numeric(otus[taxon,]) ~ MetabOrder)
  pwr <- pwr.t2n.test(n1 = length(which(meta$Metabolism == "Active")),
            n2 = length(which(meta$Metabolism == "Hibernating")), d = es$estimate)
  wil.phy <- rbind(wil.phy, data.frame(taxon = rownames(otus)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.phy$fdr <- mt(phy.phy, classlabel = "Metabolism")$fdr
#write.csv(wil.phy, "kaiju/inulinase/tax_wilcox_output/exo.phylum.output.csv")



# Class
otus <- data.frame(otu_table(cla.phy))
wil.cla <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(cla.phy)))[1]){
  es <- cohen.d(as.numeric(otus[taxon,]) ~ MetabOrder)
  pwr <- pwr.t2n.test(n1 = length(which(meta$Metabolism == "Active")),
            n2 = length(which(meta$Metabolism == "Hibernating")), d = es$estimate)
  wil.cla <- rbind(wil.cla, data.frame(taxon = rownames(otus)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.cla$fdr <- mt(cla.phy, classlabel = "Metabolism")$fdr
#write.csv(wil.cla, "kaiju/inulinase/tax_wilcox_output/exo.class.output.csv")



# Order
otus <- data.frame(otu_table(ord.phy))
wil.ord <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(ord.phy)))[1]){
  es <- cohen.d(as.numeric(otus[taxon,]) ~ MetabOrder)
  pwr <- pwr.t2n.test(n1 = length(which(meta$Metabolism == "Active")),
            n2 = length(which(meta$Metabolism == "Hibernating")), es$estimate)
  wil.ord <- rbind(wil.ord, data.frame(taxon = rownames(otus)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.ord$fdr <- mt(ord.phy, classlabel = "Metabolism")$fdr
#write.csv(wil.ord, "kaiju/inulinase/tax_wilcox_output/exo.order.output.csv")



# Family
otus <- data.frame(otu_table(fam.phy))
wil.fam <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(fam.phy)))[1]){
  es <- cohen.d(as.numeric(otus[taxon,]) ~ MetabOrder)
  pwr <- pwr.t2n.test(n1 = length(which(meta$Metabolism == "Active")),
            n2 = length(which(meta$Metabolism == "Hibernating")), d = es$estimate)
  wil.fam <- rbind(wil.fam, data.frame(taxon = rownames(otus)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.fam$fdr <- mt(fam.phy, classlabel = "Metabolism")$fdr
#write.csv(wil.fam, "kaiju/inulinase/tax_wilcox_output/exo.family.output.csv")



# Genus
otus <- data.frame(otu_table(gen.phy))
wil.gen <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(gen.phy)))[1]){
  es <- cohen.d(as.numeric(otus[taxon,]) ~ MetabOrder)
  pwr <- pwr.t2n.test(n1 = length(which(meta$Metabolism == "Active")),
            n2 = length(which(meta$Metabolism == "Hibernating")), d = es$estimate)
  wil.gen <- rbind(wil.gen, data.frame(taxon = rownames(otus)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.gen$fdr <- mt(gen.phy, classlabel = "Metabolism")$fdr
#write.csv(wil.gen, "kaiju/inulinase/tax_wilcox_output/exo.genus.output.csv")



# Species
otus <- data.frame(otu_table(spe.phy))
wil.spe <- data.frame()
for(taxon in 1:dim(data.frame(otu_table(spe.phy)))[1]){
  es <- cohen.d(as.numeric(otus[taxon,]) ~ MetabOrder)
  pwr <- pwr.t2n.test(n1 = length(which(meta$Metabolism == "Active")),
            n2 = length(which(meta$Metabolism == "Hibernating")), d = es$estimate)
  wil.spe <- rbind(wil.spe, data.frame(taxon = rownames(otus)[taxon], cohenD = es$estimate, power = pwr[[5]]))
}
wil.spe$fdr <- mt(spe.phy, classlabel = "Metabolism")$fdr
#write.csv(wil.spe, "kaiju/inulinase/tax_wilcox_output/exo.species.output.csv")

```


### Plot - Taxa
```{r}
### Phyla
phy.df <- phy.melt[-which(phy.melt$OTU == "Unclassified"),]
phy.df$OTU <- ordered(phy.df$OTU, levels=c("Bacteria;Bacteroidetes;NA;NA;NA;NA;NA;",
                                           "Bacteria;Firmicutes;NA;NA;NA;NA;NA;",
                                           "Bacteria;Actinobacteria;NA;NA;NA;NA;NA;",
                                           "Bacteria;Deinococcus-Thermus;NA;NA;NA;NA;NA;",
                                           "Eukaryota;Ascomycota;NA;NA;NA;NA;NA;",
                                           "Bacteria;Proteobacteria;NA;NA;NA;NA;NA;"))
newLabel <- strsplit(as.character(phy.df$OTU), split = ";", fixed = T)
name <- sapply(newLabel, "[", 2)
phy.df$Phylum <- name
phy.df$Phylum <- ordered(phy.df$Phylum, levels = c("Bacteroidetes", "Firmicutes", "Actinobacteria", "Deinococcus-Thermus", "Proteobacteria", "Ascomycota"))


tiff("inulinase.phyla.tiff", width = 300, height = 180, units = "mm", res = 400)
ggplot(phy.df, aes(x=Metabolism, y=Abundance, fill=Metabolism)) +
  geom_dotplot(binaxis='y', stackdir = "center", dotsize = 1.5) +
  facet_wrap(Phylum ~ ., ncol = 3, scales = "free", labeller = labeller(OTU = newLabel)) +
  scale_fill_manual(name="Seasonal\nMetabolism", values=c("white", "gray60")) +
  ylab("Relative Abundance (%)") +
  xlab("Seasonal Metabolism") +
  stat_summary(fun.data = "mean_se", fun.args = list(mult = 1), geom = "pointrange",
               color = "black", position = position_dodge(0), size = 0.75,
               show.legend = F) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.text.x = element_text(size=14),
        axis.text.y = element_text(size=14),
        axis.title.x = element_text(size=20, margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y = element_text(size=20, margin=margin(t=0, r=10, b=0, l=0)),
        strip.text.x = element_text(size=16),
        strip.text.y = element_text(size=16),
        panel.spacing=unit(1, "lines"),
        legend.title = element_text(size = 18, face="bold"),
        legend.text = element_text(size=16),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
  ) 
dev.off()







### Species
newSpe <- strsplit(as.character(spe.melt$OTU), split = ";", fixed = T)
speName <- sapply(newSpe, "[", 7)
spe.df <- spe.melt
spe.df$Species <- speName
spe.df$Species[which(spe.df$Species == NA)] <- spe.df$ta7[which(spe.df$Species == NA)]

speUse <- c("Candidatus Amulumruptor caecigallinarius", 
            #"Muribaculaceae bacterium Isolate-002 (NCI)", 
            #"Muribaculaceae bacterium Isolate-013 (NCI)", 
            "Bacteroides heparinolyticus", 
            #"Prevotella sp. ne3005", 
            "Bacteroides acidifaciens")
speUse.plot <- data.frame()

for(i in 1:length(speUse)){
  useRow <- which(spe.df$Species == speUse[i])
  speUse.plot <- rbind(speUse.plot, spe.df[useRow,])
}

speUse.plot$Species <- ordered(speUse.plot$Species, levels=c("Candidatus Amulumruptor caecigallinarius", 
                                                             #"Muribaculaceae bacterium Isolate-002 (NCI)", 
                                                             #"Muribaculaceae bacterium Isolate-013 (NCI)", 
                                                             "Bacteroides heparinolyticus", 
                                                             #"Prevotella sp. ne3005", 
                                                             "Bacteroides acidifaciens"))


#tiff("inulinase.species.tiff", width = 410, height = 100, units = "mm", res = 400)
ggplot(speUse.plot, aes(x=Metabolism, y=Abundance, fill=Metabolism)) +
  geom_dotplot(binaxis='y', stackdir = "center", dotsize = 1.5) +
  facet_wrap(Species ~ ., ncol = 3, scales = "free") +
  scale_fill_manual(name="Seasonal\nMetabolism", values=c("white", "gray60")) +
  ylab("Relative Abundance (%)") +
  xlab("Seasonal Metabolism") +
  stat_summary(fun.data = "mean_se", fun.args = list(mult = 1), geom = "pointrange",
               color = "black", position = position_dodge(0), size = 0.75,
               show.legend = F) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.text.x = element_text(size=14),
        axis.text.y = element_text(size=14),
        axis.title.x = element_text(size=20, margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y = element_text(size=20, margin=margin(t=0, r=10, b=0, l=0)),
        strip.text.x = element_text(size=16),
        strip.text.y = element_text(size=16),
        panel.spacing=unit(1, "lines"),
        legend.title = element_text(size = 18, face="bold"),
        legend.text = element_text(size=16),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
  ) 
#dev.off()






newSpe <- strsplit(as.character(spe.melt$OTU), split = ";", fixed = T)
speName <- sapply(newSpe, "[", 7)
spe.df <- spe.melt
spe.df$Species <- speName
spe.df$Species[which(spe.df$Species == NA)] <- spe.df$ta7[which(spe.df$Species == NA)]

speUse2 <- c("Aspergillus mulundensis", "Talaromyces amestolkiae")
speUse2.plot <- data.frame()

for(i in 1:length(speUse2)){
  useRow2 <- which(spe.df$Species == speUse2[i])
  speUse2.plot <- rbind(speUse2.plot, spe.df[useRow2,])
}

speUse2.plot$Species <- ordered(speUse2.plot$Species, levels=c("Aspergillus mulundensis",
                                                              "Talaromyces amestolkiae"))


tiff("inulinase.species.fungi.tiff", width = 215, height = 100, units = "mm", res = 400)
ggplot(speUse2.plot, aes(x=Metabolism, y=Abundance, fill=Metabolism)) +
  geom_dotplot(binaxis='y', stackdir = "center", dotsize = 1.5) +
  facet_wrap(Species ~ ., ncol = 3, scales = "free") +
  scale_fill_manual(name="Seasonal\nMetabolism", values=c("white", "gray60")) +
  ylab("Relative Abundance (%)") +
  xlab("Seasonal Metabolism") +
  stat_summary(fun.data = "mean_se", fun.args = list(mult = 1), geom = "pointrange",
               color = "black", position = position_dodge(0), size = 0.75,
               show.legend = F) +
  theme(plot.title=element_text(hjust=0.5, size=20),
        axis.text.x = element_text(size=14),
        axis.text.y = element_text(size=14),
        axis.title.x = element_text(size=20, margin=margin(t=10, r=0, b=0, l=0)),
        axis.title.y = element_text(size=20, margin=margin(t=0, r=10, b=0, l=0)),
        strip.text.x = element_text(size=16),
        strip.text.y = element_text(size=16),
        panel.spacing=unit(1, "lines"),
        legend.title = element_text(size = 18, face="bold"),
        legend.text = element_text(size=16),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        strip.background = element_blank(),
  ) 
dev.off()





```